{
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "XUZS4GUkHIEG"
      },
      "source": [
        "# Подготовка файлов и программ\n",
        "\n",
        "Некоторые программы ставятся дважды из разных источников для демонстрации различных вариантов в случае неполадок. Для оптимизации времени работы необходимо удалить лишний код."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 2,
      "metadata": {
        "id": "lrtRbP3fAvj8"
      },
      "outputs": [],
      "source": [
        "pip install -q condacolab"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 3,
      "metadata": {
        "id": "jQ1Oqzb9BXoG"
      },
      "outputs": [],
      "source": [
        "import condacolab"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 4,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "q5FUOHCgBZWp",
        "outputId": "3ca19346-9ff8-4514-f28c-c47983004b05"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "⏬ Downloading https://github.com/jaimergp/miniforge/releases/latest/download/Mambaforge-colab-Linux-x86_64.sh...\n",
            "📦 Installing...\n",
            "📌 Adjusting configuration...\n",
            "🩹 Patching environment...\n",
            "⏲ Done in 0:00:35\n",
            "🔁 Restarting kernel...\n"
          ]
        }
      ],
      "source": [
        "condacolab.install()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 1,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "XnbIt77YBOT9",
        "outputId": "c0189d41-9429-44d8-a290-3d4ee67bb9a1"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\bdone\n",
            "Solving environment: \\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\bdone\n",
            "\n",
            "## Package Plan ##\n",
            "\n",
            "  environment location: /usr/local\n",
            "\n",
            "  added / updated specs:\n",
            "    - seqtk\n",
            "\n",
            "\n",
            "The following packages will be downloaded:\n",
            "\n",
            "    package                    |            build\n",
            "    ---------------------------|-----------------\n",
            "    ca-certificates-2021.10.8  |       ha878542_0         139 KB  conda-forge\n",
            "    certifi-2021.10.8          |   py37h89c1867_1         145 KB  conda-forge\n",
            "    conda-4.11.0               |   py37h89c1867_2        16.9 MB  conda-forge\n",
            "    seqtk-1.3                  |       h5bf99c6_3          38 KB  bioconda\n",
            "    ------------------------------------------------------------\n",
            "                                           Total:        17.2 MB\n",
            "\n",
            "The following NEW packages will be INSTALLED:\n",
            "\n",
            "  seqtk              bioconda/linux-64::seqtk-1.3-h5bf99c6_3\n",
            "\n",
            "The following packages will be UPDATED:\n",
            "\n",
            "  ca-certificates                      2020.12.5-ha878542_0 --> 2021.10.8-ha878542_0\n",
            "  certifi                          2020.12.5-py37h89c1867_1 --> 2021.10.8-py37h89c1867_1\n",
            "  conda                                4.9.2-py37h89c1867_0 --> 4.11.0-py37h89c1867_2\n",
            "\n",
            "\n",
            "\n",
            "Downloading and Extracting Packages\n",
            "ca-certificates-2021 | 139 KB    | : 100% 1.0/1 [00:00<00:00, 10.31it/s]\n",
            "seqtk-1.3            | 38 KB     | : 100% 1.0/1 [00:00<00:00, 15.13it/s]\n",
            "conda-4.11.0         | 16.9 MB   | : 100% 1.0/1 [00:05<00:00,  5.10s/it]               \n",
            "certifi-2021.10.8    | 145 KB    | : 100% 1.0/1 [00:00<00:00, 14.27it/s]\n",
            "Preparing transaction: \\ \b\bdone\n",
            "Verifying transaction: / \b\bdone\n",
            "Executing transaction: \\ \b\bdone\n"
          ]
        }
      ],
      "source": [
        "!conda install -c bioconda seqtk"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 2,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "sCBV_31qHOik",
        "outputId": "e51a45d3-7eb5-4e5e-9dfa-1be933207593"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\bdone\n",
            "Solving environment: - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\bdone\n",
            "\n",
            "## Package Plan ##\n",
            "\n",
            "  environment location: /usr/local\n",
            "\n",
            "  added / updated specs:\n",
            "    - trimmomatic\n",
            "\n",
            "\n",
            "The following packages will be downloaded:\n",
            "\n",
            "    package                    |            build\n",
            "    ---------------------------|-----------------\n",
            "    openjdk-8.0.312            |       h7f98852_0        97.6 MB  conda-forge\n",
            "    python_abi-3.7             |          2_cp37m           4 KB  conda-forge\n",
            "    trimmomatic-0.39           |       hdfd78af_2         144 KB  bioconda\n",
            "    ------------------------------------------------------------\n",
            "                                           Total:        97.8 MB\n",
            "\n",
            "The following NEW packages will be INSTALLED:\n",
            "\n",
            "  openjdk            conda-forge/linux-64::openjdk-8.0.312-h7f98852_0\n",
            "  trimmomatic        bioconda/noarch::trimmomatic-0.39-hdfd78af_2\n",
            "\n",
            "The following packages will be UPDATED:\n",
            "\n",
            "  python_abi                                    3.7-1_cp37m --> 3.7-2_cp37m\n",
            "\n",
            "\n",
            "\n",
            "Downloading and Extracting Packages\n",
            "trimmomatic-0.39     | 144 KB    | : 100% 1.0/1 [00:00<00:00, 13.55it/s]\n",
            "python_abi-3.7       | 4 KB      | : 100% 1.0/1 [00:00<00:00, 17.29it/s]\n",
            "openjdk-8.0.312      | 97.6 MB   | : 100% 1.0/1 [00:17<00:00, 17.04s/it]               \n",
            "Preparing transaction: \\ \b\bdone\n",
            "Verifying transaction: / \b\bdone\n",
            "Executing transaction: \\ \b\bdone\n"
          ]
        }
      ],
      "source": [
        "!conda install -c bioconda trimmomatic"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 3,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        },
        "id": "F0BLWsH2n3Ov",
        "outputId": "0c2b1617-8c2a-49f6-9ffb-70a872b3be72"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Collecting macs2\n",
            "  Downloading MACS2-2.2.7.1.tar.gz (2.1 MB)\n",
            "\u001b[?25l\r\u001b[K     |▏                               | 10 kB 25.2 MB/s eta 0:00:01\r\u001b[K     |▎                               | 20 kB 31.6 MB/s eta 0:00:01\r\u001b[K     |▌                               | 30 kB 32.8 MB/s eta 0:00:01\r\u001b[K     |▋                               | 40 kB 28.7 MB/s eta 0:00:01\r\u001b[K     |▉                               | 51 kB 22.4 MB/s eta 0:00:01\r\u001b[K     |█                               | 61 kB 25.4 MB/s eta 0:00:01\r\u001b[K     |█▏                              | 71 kB 22.3 MB/s eta 0:00:01\r\u001b[K     |█▎                              | 81 kB 23.3 MB/s eta 0:00:01\r\u001b[K     |█▍                              | 92 kB 25.4 MB/s eta 0:00:01\r\u001b[K     |█▋                              | 102 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |█▊                              | 112 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |██                              | 122 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |██                              | 133 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |██▎                             | 143 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |██▍                             | 153 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |██▌                             | 163 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |██▊                             | 174 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |██▉                             | 184 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |███                             | 194 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |███▏                            | 204 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |███▍                            | 215 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |███▌                            | 225 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |███▋                            | 235 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |███▉                            | 245 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |████                            | 256 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |████▏                           | 266 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |████▎                           | 276 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |████▌                           | 286 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |████▋                           | 296 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |████▊                           | 307 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |█████                           | 317 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |█████                           | 327 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |█████▎                          | 337 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |█████▍                          | 348 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |█████▋                          | 358 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |█████▊                          | 368 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |██████                          | 378 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |██████                          | 389 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |██████▏                         | 399 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |██████▍                         | 409 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |██████▌                         | 419 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |██████▊                         | 430 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |██████▉                         | 440 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |███████                         | 450 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |███████▏                        | 460 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |███████▎                        | 471 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |███████▌                        | 481 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |███████▋                        | 491 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |███████▉                        | 501 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |████████                        | 512 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |████████▏                       | 522 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |████████▎                       | 532 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |████████▍                       | 542 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |████████▋                       | 552 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |████████▊                       | 563 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |█████████                       | 573 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |█████████                       | 583 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |█████████▎                      | 593 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |█████████▍                      | 604 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |█████████▌                      | 614 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |█████████▊                      | 624 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |█████████▉                      | 634 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |██████████                      | 645 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |██████████▏                     | 655 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |██████████▍                     | 665 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |██████████▌                     | 675 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |██████████▊                     | 686 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |██████████▉                     | 696 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |███████████                     | 706 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |███████████▏                    | 716 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |███████████▎                    | 727 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |███████████▌                    | 737 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |███████████▋                    | 747 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |███████████▉                    | 757 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |████████████                    | 768 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |████████████                    | 778 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |████████████▎                   | 788 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |████████████▍                   | 798 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |████████████▋                   | 808 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |████████████▊                   | 819 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |█████████████                   | 829 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |█████████████                   | 839 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |█████████████▏                  | 849 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |█████████████▍                  | 860 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |█████████████▌                  | 870 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |█████████████▊                  | 880 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |█████████████▉                  | 890 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |██████████████                  | 901 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |██████████████▏                 | 911 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |██████████████▎                 | 921 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |██████████████▌                 | 931 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |██████████████▋                 | 942 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |██████████████▉                 | 952 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |███████████████                 | 962 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |███████████████▏                | 972 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |███████████████▎                | 983 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |███████████████▌                | 993 kB 24.5 MB/s eta 0:00:01\r\u001b[K     |███████████████▋                | 1.0 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |███████████████▊                | 1.0 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |████████████████                | 1.0 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |████████████████                | 1.0 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |████████████████▎               | 1.0 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |████████████████▍               | 1.1 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |████████████████▋               | 1.1 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |████████████████▊               | 1.1 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |████████████████▉               | 1.1 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |█████████████████               | 1.1 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |█████████████████▏              | 1.1 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |█████████████████▍              | 1.1 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |█████████████████▌              | 1.1 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |█████████████████▊              | 1.1 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |█████████████████▉              | 1.1 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |██████████████████              | 1.2 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |██████████████████▏             | 1.2 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |██████████████████▎             | 1.2 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |██████████████████▌             | 1.2 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |██████████████████▋             | 1.2 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |██████████████████▉             | 1.2 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |███████████████████             | 1.2 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |███████████████████             | 1.2 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |███████████████████▎            | 1.2 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |███████████████████▍            | 1.2 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |███████████████████▋            | 1.3 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |███████████████████▊            | 1.3 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |████████████████████            | 1.3 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |████████████████████            | 1.3 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |████████████████████▎           | 1.3 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |████████████████████▍           | 1.3 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |████████████████████▌           | 1.3 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |████████████████████▊           | 1.3 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |████████████████████▉           | 1.3 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |█████████████████████           | 1.4 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |█████████████████████▏          | 1.4 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |█████████████████████▍          | 1.4 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |█████████████████████▌          | 1.4 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |█████████████████████▋          | 1.4 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |█████████████████████▉          | 1.4 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |██████████████████████          | 1.4 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |██████████████████████▏         | 1.4 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |██████████████████████▎         | 1.4 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |██████████████████████▌         | 1.4 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |██████████████████████▋         | 1.5 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |██████████████████████▊         | 1.5 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |███████████████████████         | 1.5 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |███████████████████████         | 1.5 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |███████████████████████▎        | 1.5 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |███████████████████████▍        | 1.5 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |███████████████████████▋        | 1.5 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |███████████████████████▊        | 1.5 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |███████████████████████▉        | 1.5 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |████████████████████████        | 1.5 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |████████████████████████▏       | 1.6 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |████████████████████████▍       | 1.6 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |████████████████████████▌       | 1.6 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |████████████████████████▊       | 1.6 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |████████████████████████▉       | 1.6 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |█████████████████████████       | 1.6 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |█████████████████████████▏      | 1.6 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |█████████████████████████▎      | 1.6 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |█████████████████████████▌      | 1.6 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |█████████████████████████▋      | 1.6 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |█████████████████████████▉      | 1.7 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |██████████████████████████      | 1.7 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |██████████████████████████▏     | 1.7 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |██████████████████████████▎     | 1.7 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |██████████████████████████▍     | 1.7 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |██████████████████████████▋     | 1.7 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |██████████████████████████▊     | 1.7 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |███████████████████████████     | 1.7 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |███████████████████████████     | 1.7 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |███████████████████████████▎    | 1.8 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |███████████████████████████▍    | 1.8 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |███████████████████████████▌    | 1.8 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |███████████████████████████▊    | 1.8 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |███████████████████████████▉    | 1.8 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |████████████████████████████    | 1.8 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |████████████████████████████▏   | 1.8 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |████████████████████████████▍   | 1.8 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |████████████████████████████▌   | 1.8 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |████████████████████████████▋   | 1.8 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |████████████████████████████▉   | 1.9 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |█████████████████████████████   | 1.9 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |█████████████████████████████▏  | 1.9 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |█████████████████████████████▎  | 1.9 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |█████████████████████████████▌  | 1.9 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |█████████████████████████████▋  | 1.9 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |█████████████████████████████▊  | 1.9 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |██████████████████████████████  | 1.9 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |██████████████████████████████  | 1.9 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |██████████████████████████████▎ | 1.9 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |██████████████████████████████▍ | 2.0 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |██████████████████████████████▋ | 2.0 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |██████████████████████████████▊ | 2.0 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |███████████████████████████████ | 2.0 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |███████████████████████████████ | 2.0 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |███████████████████████████████▏| 2.0 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |███████████████████████████████▍| 2.0 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |███████████████████████████████▌| 2.0 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |███████████████████████████████▊| 2.0 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |███████████████████████████████▉| 2.0 MB 24.5 MB/s eta 0:00:01\r\u001b[K     |████████████████████████████████| 2.1 MB 24.5 MB/s \n",
            "\u001b[?25h  Installing build dependencies ... \u001b[?25l\u001b[?25hdone\n",
            "  Getting requirements to build wheel ... \u001b[?25l\u001b[?25hdone\n",
            "    Preparing wheel metadata ... \u001b[?25l\u001b[?25hdone\n",
            "Collecting intervene\n",
            "  Downloading intervene-0.6.5.tar.gz (5.5 MB)\n",
            "\u001b[K     |████████████████████████████████| 5.5 MB 50.8 MB/s \n",
            "\u001b[?25hCollecting pybedtools\n",
            "  Downloading pybedtools-0.9.0.tar.gz (12.5 MB)\n",
            "\u001b[K     |████████████████████████████████| 12.5 MB 32.6 MB/s \n",
            "\u001b[?25hCollecting matplotlib\n",
            "  Downloading matplotlib-3.5.1-cp37-cp37m-manylinux_2_5_x86_64.manylinux1_x86_64.whl (11.2 MB)\n",
            "\u001b[K     |████████████████████████████████| 11.2 MB 65.3 MB/s \n",
            "\u001b[?25hCollecting pandas\n",
            "  Downloading pandas-1.3.5-cp37-cp37m-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (11.3 MB)\n",
            "\u001b[K     |████████████████████████████████| 11.3 MB 46.0 MB/s \n",
            "\u001b[?25hCollecting numpy\n",
            "  Using cached numpy-1.21.5-cp37-cp37m-manylinux_2_12_x86_64.manylinux2010_x86_64.whl (15.7 MB)\n",
            "Collecting scipy\n",
            "  Downloading scipy-1.7.3-cp37-cp37m-manylinux_2_12_x86_64.manylinux2010_x86_64.whl (38.1 MB)\n",
            "\u001b[K     |████████████████████████████████| 38.1 MB 1.3 MB/s \n",
            "\u001b[?25hCollecting seaborn\n",
            "  Downloading seaborn-0.11.2-py3-none-any.whl (292 kB)\n",
            "\u001b[K     |████████████████████████████████| 292 kB 45.9 MB/s \n",
            "\u001b[?25hCollecting pyparsing>=2.2.1\n",
            "  Downloading pyparsing-3.0.7-py3-none-any.whl (98 kB)\n",
            "\u001b[K     |████████████████████████████████| 98 kB 7.1 MB/s \n",
            "\u001b[?25hCollecting python-dateutil>=2.7\n",
            "  Downloading python_dateutil-2.8.2-py2.py3-none-any.whl (247 kB)\n",
            "\u001b[K     |████████████████████████████████| 247 kB 54.1 MB/s \n",
            "\u001b[?25hCollecting fonttools>=4.22.0\n",
            "  Downloading fonttools-4.30.0-py3-none-any.whl (898 kB)\n",
            "\u001b[K     |████████████████████████████████| 898 kB 65.9 MB/s \n",
            "\u001b[?25hCollecting kiwisolver>=1.0.1\n",
            "  Downloading kiwisolver-1.3.2-cp37-cp37m-manylinux_2_5_x86_64.manylinux1_x86_64.whl (1.1 MB)\n",
            "\u001b[K     |████████████████████████████████| 1.1 MB 58.3 MB/s \n",
            "\u001b[?25hCollecting pillow>=6.2.0\n",
            "  Downloading Pillow-9.0.1-cp37-cp37m-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (4.3 MB)\n",
            "\u001b[K     |████████████████████████████████| 4.3 MB 39.0 MB/s \n",
            "\u001b[?25hCollecting cycler>=0.10\n",
            "  Downloading cycler-0.11.0-py3-none-any.whl (6.4 kB)\n",
            "Collecting packaging>=20.0\n",
            "  Downloading packaging-21.3-py3-none-any.whl (40 kB)\n",
            "\u001b[K     |████████████████████████████████| 40 kB 5.2 MB/s \n",
            "\u001b[?25hRequirement already satisfied: six>=1.5 in /usr/local/lib/python3.7/site-packages (from python-dateutil>=2.7->matplotlib->intervene) (1.15.0)\n",
            "Collecting pytz>=2017.3\n",
            "  Downloading pytz-2021.3-py2.py3-none-any.whl (503 kB)\n",
            "\u001b[K     |████████████████████████████████| 503 kB 68.6 MB/s \n",
            "\u001b[?25hCollecting pysam\n",
            "  Downloading pysam-0.18.0-cp37-cp37m-manylinux_2_12_x86_64.manylinux2010_x86_64.whl (14.9 MB)\n",
            "\u001b[K     |████████████████████████████████| 14.9 MB 60.1 MB/s \n",
            "\u001b[?25hBuilding wheels for collected packages: intervene, macs2, pybedtools\n",
            "  Building wheel for intervene (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for intervene: filename=intervene-0.6.5-py3-none-any.whl size=5546118 sha256=741f1b7a30440664514fb260d5010cda57c8f462d1eb463c2661e6df95817503\n",
            "  Stored in directory: /root/.cache/pip/wheels/f3/d2/c1/18fa6122c36fdc731af4f16c59c356606b12a890a578294044\n",
            "  Building wheel for macs2 (PEP 517) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for macs2: filename=MACS2-2.2.7.1-cp37-cp37m-linux_x86_64.whl size=1790967 sha256=4728d5db41205f0963fb08e1559d2f337a5e28124dca44a67600cd6d64e092b8\n",
            "  Stored in directory: /root/.cache/pip/wheels/4e/34/10/f6222945f7c5b091e34b058a01a842ba9b1d960f2ab0c8e16f\n",
            "  Building wheel for pybedtools (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for pybedtools: filename=pybedtools-0.9.0-cp37-cp37m-linux_x86_64.whl size=13654297 sha256=8d380134f5e3c28ce493c4b0d13f47e817790ff7a8a8b622f2b9d0f37e11ca3b\n",
            "  Stored in directory: /root/.cache/pip/wheels/7a/44/0d/3a7449885adaf8ebb157da8c3c834a712f48b3b3b84ba51dda\n",
            "Successfully built intervene macs2 pybedtools\n",
            "Installing collected packages: pyparsing, pytz, python-dateutil, pillow, packaging, numpy, kiwisolver, fonttools, cycler, scipy, pysam, pandas, matplotlib, seaborn, pybedtools, macs2, intervene\n",
            "Successfully installed cycler-0.11.0 fonttools-4.30.0 intervene-0.6.5 kiwisolver-1.3.2 macs2-2.2.7.1 matplotlib-3.5.1 numpy-1.21.5 packaging-21.3 pandas-1.3.5 pillow-9.0.1 pybedtools-0.9.0 pyparsing-3.0.7 pysam-0.18.0 python-dateutil-2.8.2 pytz-2021.3 scipy-1.7.3 seaborn-0.11.2\n"
          ]
        },
        {
          "output_type": "display_data",
          "data": {
            "application/vnd.colab-display-data+json": {
              "pip_warning": {
                "packages": [
                  "cycler",
                  "dateutil",
                  "kiwisolver",
                  "numpy",
                  "pyparsing"
                ]
              }
            }
          },
          "metadata": {}
        }
      ],
      "source": [
        "!pip install macs2 intervene"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 4,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Ce6hiSTdkarU",
        "outputId": "b7b0dc81-8081-4409-b888-15c5797438b6"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Reading package lists... Done\n",
            "Building dependency tree       \n",
            "Reading state information... Done\n",
            "The following NEW packages will be installed:\n",
            "  bedtools\n",
            "0 upgraded, 1 newly installed, 0 to remove and 39 not upgraded.\n",
            "Need to get 577 kB of archives.\n",
            "After this operation, 2,040 kB of additional disk space will be used.\n",
            "Get:1 http://archive.ubuntu.com/ubuntu bionic/universe amd64 bedtools amd64 2.26.0+dfsg-5 [577 kB]\n",
            "Fetched 577 kB in 1s (967 kB/s)\n",
            "Selecting previously unselected package bedtools.\n",
            "(Reading database ... 155335 files and directories currently installed.)\n",
            "Preparing to unpack .../bedtools_2.26.0+dfsg-5_amd64.deb ...\n",
            "Unpacking bedtools (2.26.0+dfsg-5) ...\n",
            "Setting up bedtools (2.26.0+dfsg-5) ...\n"
          ]
        }
      ],
      "source": [
        "!apt-get install bedtools"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 5,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "AUML3dvsqT9E",
        "outputId": "0aa297fb-7723-4605-b606-eb1d6491c366"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Reading package lists... Done\n",
            "Building dependency tree       \n",
            "Reading state information... Done\n",
            "The following additional packages will be installed:\n",
            "  cwltool libc-ares2 libhts2 libjs-bootstrap nodejs nodejs-doc\n",
            "  python-asn1crypto python-avro python-cachecontrol python-certifi\n",
            "  python-cffi-backend python-chardet python-cryptography python-enum34\n",
            "  python-html5lib python-idna python-ipaddress python-isodate python-lockfile\n",
            "  python-mistune python-openssl python-pkg-resources python-pyparsing\n",
            "  python-rdflib python-rdflib-jsonld python-requests python-ruamel.yaml\n",
            "  python-schema-salad python-shellescape python-six python-sparqlwrapper\n",
            "  python-typing python-urllib3 python-webencodings\n",
            "Suggested packages:\n",
            "  bowtie2-examples python-cryptography-doc python-cryptography-vectors\n",
            "  python-enum34-doc python-genshi python-lxml python-lockfile-doc\n",
            "  python-openssl-doc python-openssl-dbg python-setuptools python-pyparsing-doc\n",
            "  python-rdflib-doc python-rdflib-tools python-socks python-ntlm\n",
            "The following NEW packages will be installed:\n",
            "  bowtie2 cwltool libc-ares2 libhts2 libjs-bootstrap nodejs nodejs-doc\n",
            "  python-asn1crypto python-avro python-cachecontrol python-certifi\n",
            "  python-cffi-backend python-chardet python-cryptography python-enum34\n",
            "  python-html5lib python-idna python-ipaddress python-isodate python-lockfile\n",
            "  python-mistune python-openssl python-pkg-resources python-pyparsing\n",
            "  python-rdflib python-rdflib-jsonld python-requests python-ruamel.yaml\n",
            "  python-schema-salad python-shellescape python-six python-sparqlwrapper\n",
            "  python-typing python-urllib3 python-webencodings samtools\n",
            "0 upgraded, 36 newly installed, 0 to remove and 39 not upgraded.\n",
            "Need to get 9,743 kB of archives.\n",
            "After this operation, 46.1 MB of additional disk space will be used.\n",
            "Get:1 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-pkg-resources all 39.0.1-2 [128 kB]\n",
            "Get:2 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-six all 1.11.0-2 [11.3 kB]\n",
            "Get:3 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-isodate all 0.6.0-1 [23.8 kB]\n",
            "Get:4 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-pyparsing all 2.2.0+dfsg1-2 [52.1 kB]\n",
            "Get:5 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-rdflib all 4.2.1-2 [237 kB]\n",
            "Get:6 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-certifi all 2018.1.18-2 [144 kB]\n",
            "Get:7 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-chardet all 3.0.4-1 [80.3 kB]\n",
            "Get:8 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-idna all 2.6-1 [32.4 kB]\n",
            "Get:9 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 python-urllib3 all 1.22-1ubuntu0.18.04.2 [86.0 kB]\n",
            "Get:10 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 python-requests all 2.18.4-2ubuntu0.1 [58.5 kB]\n",
            "Get:11 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-ruamel.yaml amd64 0.15.34-1 [183 kB]\n",
            "Get:12 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-cachecontrol all 0.11.7-1 [15.2 kB]\n",
            "Get:13 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-lockfile all 1:0.12.2-2 [14.6 kB]\n",
            "Get:14 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-mistune all 0.8.3-2 [13.8 kB]\n",
            "Get:15 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-rdflib-jsonld all 0.4.0-3 [12.2 kB]\n",
            "Get:16 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-typing all 3.6.2-1 [22.2 kB]\n",
            "Get:17 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-avro all 1.8.2+dfsg-1 [23.2 kB]\n",
            "Get:18 http://archive.ubuntu.com/ubuntu bionic/universe amd64 libjs-bootstrap all 3.3.7+dfsg-2 [228 kB]\n",
            "Get:19 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-schema-salad all 2.6.20171201034858-3 [260 kB]\n",
            "Get:20 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-shellescape all 3.4.1-2 [4,432 B]\n",
            "Get:21 http://archive.ubuntu.com/ubuntu bionic/universe amd64 cwltool all 1.0.20180302231433-1 [143 kB]\n",
            "Get:22 http://archive.ubuntu.com/ubuntu bionic/universe amd64 libhts2 amd64 1.7-2 [300 kB]\n",
            "Get:23 http://archive.ubuntu.com/ubuntu bionic-updates/universe amd64 nodejs-doc all 8.10.0~dfsg-2ubuntu0.4 [752 kB]\n",
            "Get:24 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-asn1crypto all 0.24.0-1 [72.7 kB]\n",
            "Get:25 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-cffi-backend amd64 1.11.5-1 [63.4 kB]\n",
            "Get:26 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-enum34 all 1.1.6-2 [34.8 kB]\n",
            "Get:27 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-ipaddress all 1.0.17-1 [18.2 kB]\n",
            "Get:28 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 python-cryptography amd64 2.1.4-1ubuntu1.4 [276 kB]\n",
            "Get:29 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-webencodings all 0.5-2 [10.3 kB]\n",
            "Get:30 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-html5lib all 0.999999999-1 [83.6 kB]\n",
            "Get:31 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-openssl all 17.5.0-1ubuntu1 [41.3 kB]\n",
            "Get:32 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-sparqlwrapper all 1.7.6-2 [21.8 kB]\n",
            "Get:33 http://archive.ubuntu.com/ubuntu bionic/universe amd64 samtools amd64 1.7-1 [288 kB]\n",
            "Get:34 http://archive.ubuntu.com/ubuntu bionic/universe amd64 bowtie2 amd64 2.3.4.1-1 [1,174 kB]\n",
            "Get:35 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 libc-ares2 amd64 1.14.0-1ubuntu0.1 [37.5 kB]\n",
            "Get:36 http://archive.ubuntu.com/ubuntu bionic-updates/universe amd64 nodejs amd64 8.10.0~dfsg-2ubuntu0.4 [4,796 kB]\n",
            "Fetched 9,743 kB in 1s (7,855 kB/s)\n",
            "Extracting templates from packages: 100%\n",
            "Selecting previously unselected package python-pkg-resources.\n",
            "(Reading database ... 155386 files and directories currently installed.)\n",
            "Preparing to unpack .../00-python-pkg-resources_39.0.1-2_all.deb ...\n",
            "Unpacking python-pkg-resources (39.0.1-2) ...\n",
            "Selecting previously unselected package python-six.\n",
            "Preparing to unpack .../01-python-six_1.11.0-2_all.deb ...\n",
            "Unpacking python-six (1.11.0-2) ...\n",
            "Selecting previously unselected package python-isodate.\n",
            "Preparing to unpack .../02-python-isodate_0.6.0-1_all.deb ...\n",
            "Unpacking python-isodate (0.6.0-1) ...\n",
            "Selecting previously unselected package python-pyparsing.\n",
            "Preparing to unpack .../03-python-pyparsing_2.2.0+dfsg1-2_all.deb ...\n",
            "Unpacking python-pyparsing (2.2.0+dfsg1-2) ...\n",
            "Selecting previously unselected package python-rdflib.\n",
            "Preparing to unpack .../04-python-rdflib_4.2.1-2_all.deb ...\n",
            "Unpacking python-rdflib (4.2.1-2) ...\n",
            "Selecting previously unselected package python-certifi.\n",
            "Preparing to unpack .../05-python-certifi_2018.1.18-2_all.deb ...\n",
            "Unpacking python-certifi (2018.1.18-2) ...\n",
            "Selecting previously unselected package python-chardet.\n",
            "Preparing to unpack .../06-python-chardet_3.0.4-1_all.deb ...\n",
            "Unpacking python-chardet (3.0.4-1) ...\n",
            "Selecting previously unselected package python-idna.\n",
            "Preparing to unpack .../07-python-idna_2.6-1_all.deb ...\n",
            "Unpacking python-idna (2.6-1) ...\n",
            "Selecting previously unselected package python-urllib3.\n",
            "Preparing to unpack .../08-python-urllib3_1.22-1ubuntu0.18.04.2_all.deb ...\n",
            "Unpacking python-urllib3 (1.22-1ubuntu0.18.04.2) ...\n",
            "Selecting previously unselected package python-requests.\n",
            "Preparing to unpack .../09-python-requests_2.18.4-2ubuntu0.1_all.deb ...\n",
            "Unpacking python-requests (2.18.4-2ubuntu0.1) ...\n",
            "Selecting previously unselected package python-ruamel.yaml.\n",
            "Preparing to unpack .../10-python-ruamel.yaml_0.15.34-1_amd64.deb ...\n",
            "Unpacking python-ruamel.yaml (0.15.34-1) ...\n",
            "Selecting previously unselected package python-cachecontrol.\n",
            "Preparing to unpack .../11-python-cachecontrol_0.11.7-1_all.deb ...\n",
            "Unpacking python-cachecontrol (0.11.7-1) ...\n",
            "Selecting previously unselected package python-lockfile.\n",
            "Preparing to unpack .../12-python-lockfile_1%3a0.12.2-2_all.deb ...\n",
            "Unpacking python-lockfile (1:0.12.2-2) ...\n",
            "Selecting previously unselected package python-mistune.\n",
            "Preparing to unpack .../13-python-mistune_0.8.3-2_all.deb ...\n",
            "Unpacking python-mistune (0.8.3-2) ...\n",
            "Selecting previously unselected package python-rdflib-jsonld.\n",
            "Preparing to unpack .../14-python-rdflib-jsonld_0.4.0-3_all.deb ...\n",
            "Unpacking python-rdflib-jsonld (0.4.0-3) ...\n",
            "Selecting previously unselected package python-typing.\n",
            "Preparing to unpack .../15-python-typing_3.6.2-1_all.deb ...\n",
            "Unpacking python-typing (3.6.2-1) ...\n",
            "Selecting previously unselected package python-avro.\n",
            "Preparing to unpack .../16-python-avro_1.8.2+dfsg-1_all.deb ...\n",
            "Unpacking python-avro (1.8.2+dfsg-1) ...\n",
            "Selecting previously unselected package libjs-bootstrap.\n",
            "Preparing to unpack .../17-libjs-bootstrap_3.3.7+dfsg-2_all.deb ...\n",
            "Unpacking libjs-bootstrap (3.3.7+dfsg-2) ...\n",
            "Selecting previously unselected package python-schema-salad.\n",
            "Preparing to unpack .../18-python-schema-salad_2.6.20171201034858-3_all.deb ...\n",
            "Unpacking python-schema-salad (2.6.20171201034858-3) ...\n",
            "Selecting previously unselected package python-shellescape.\n",
            "Preparing to unpack .../19-python-shellescape_3.4.1-2_all.deb ...\n",
            "Unpacking python-shellescape (3.4.1-2) ...\n",
            "Selecting previously unselected package cwltool.\n",
            "Preparing to unpack .../20-cwltool_1.0.20180302231433-1_all.deb ...\n",
            "Unpacking cwltool (1.0.20180302231433-1) ...\n",
            "Selecting previously unselected package libhts2:amd64.\n",
            "Preparing to unpack .../21-libhts2_1.7-2_amd64.deb ...\n",
            "Unpacking libhts2:amd64 (1.7-2) ...\n",
            "Selecting previously unselected package nodejs-doc.\n",
            "Preparing to unpack .../22-nodejs-doc_8.10.0~dfsg-2ubuntu0.4_all.deb ...\n",
            "Unpacking nodejs-doc (8.10.0~dfsg-2ubuntu0.4) ...\n",
            "Selecting previously unselected package python-asn1crypto.\n",
            "Preparing to unpack .../23-python-asn1crypto_0.24.0-1_all.deb ...\n",
            "Unpacking python-asn1crypto (0.24.0-1) ...\n",
            "Selecting previously unselected package python-cffi-backend.\n",
            "Preparing to unpack .../24-python-cffi-backend_1.11.5-1_amd64.deb ...\n",
            "Unpacking python-cffi-backend (1.11.5-1) ...\n",
            "Selecting previously unselected package python-enum34.\n",
            "Preparing to unpack .../25-python-enum34_1.1.6-2_all.deb ...\n",
            "Unpacking python-enum34 (1.1.6-2) ...\n",
            "Selecting previously unselected package python-ipaddress.\n",
            "Preparing to unpack .../26-python-ipaddress_1.0.17-1_all.deb ...\n",
            "Unpacking python-ipaddress (1.0.17-1) ...\n",
            "Selecting previously unselected package python-cryptography.\n",
            "Preparing to unpack .../27-python-cryptography_2.1.4-1ubuntu1.4_amd64.deb ...\n",
            "Unpacking python-cryptography (2.1.4-1ubuntu1.4) ...\n",
            "Selecting previously unselected package python-webencodings.\n",
            "Preparing to unpack .../28-python-webencodings_0.5-2_all.deb ...\n",
            "Unpacking python-webencodings (0.5-2) ...\n",
            "Selecting previously unselected package python-html5lib.\n",
            "Preparing to unpack .../29-python-html5lib_0.999999999-1_all.deb ...\n",
            "Unpacking python-html5lib (0.999999999-1) ...\n",
            "Selecting previously unselected package python-openssl.\n",
            "Preparing to unpack .../30-python-openssl_17.5.0-1ubuntu1_all.deb ...\n",
            "Unpacking python-openssl (17.5.0-1ubuntu1) ...\n",
            "Selecting previously unselected package python-sparqlwrapper.\n",
            "Preparing to unpack .../31-python-sparqlwrapper_1.7.6-2_all.deb ...\n",
            "Unpacking python-sparqlwrapper (1.7.6-2) ...\n",
            "Selecting previously unselected package samtools.\n",
            "Preparing to unpack .../32-samtools_1.7-1_amd64.deb ...\n",
            "Unpacking samtools (1.7-1) ...\n",
            "Selecting previously unselected package bowtie2.\n",
            "Preparing to unpack .../33-bowtie2_2.3.4.1-1_amd64.deb ...\n",
            "Unpacking bowtie2 (2.3.4.1-1) ...\n",
            "Selecting previously unselected package libc-ares2:amd64.\n",
            "Preparing to unpack .../34-libc-ares2_1.14.0-1ubuntu0.1_amd64.deb ...\n",
            "Unpacking libc-ares2:amd64 (1.14.0-1ubuntu0.1) ...\n",
            "Selecting previously unselected package nodejs.\n",
            "Preparing to unpack .../35-nodejs_8.10.0~dfsg-2ubuntu0.4_amd64.deb ...\n",
            "Unpacking nodejs (8.10.0~dfsg-2ubuntu0.4) ...\n",
            "Setting up python-idna (2.6-1) ...\n",
            "Setting up nodejs-doc (8.10.0~dfsg-2ubuntu0.4) ...\n",
            "Setting up python-asn1crypto (0.24.0-1) ...\n",
            "Setting up python-mistune (0.8.3-2) ...\n",
            "Setting up python-certifi (2018.1.18-2) ...\n",
            "Setting up python-avro (1.8.2+dfsg-1) ...\n",
            "Setting up python-pkg-resources (39.0.1-2) ...\n",
            "Setting up bowtie2 (2.3.4.1-1) ...\n",
            "Setting up python-ruamel.yaml (0.15.34-1) ...\n",
            "Setting up python-cffi-backend (1.11.5-1) ...\n",
            "Setting up python-six (1.11.0-2) ...\n",
            "Setting up python-isodate (0.6.0-1) ...\n",
            "Setting up python-shellescape (3.4.1-2) ...\n",
            "Setting up python-enum34 (1.1.6-2) ...\n",
            "Setting up libjs-bootstrap (3.3.7+dfsg-2) ...\n",
            "Setting up python-lockfile (1:0.12.2-2) ...\n",
            "Setting up python-ipaddress (1.0.17-1) ...\n",
            "Setting up libc-ares2:amd64 (1.14.0-1ubuntu0.1) ...\n",
            "Setting up libhts2:amd64 (1.7-2) ...\n",
            "Setting up python-webencodings (0.5-2) ...\n",
            "Setting up python-typing (3.6.2-1) ...\n",
            "Setting up python-pyparsing (2.2.0+dfsg1-2) ...\n",
            "Setting up python-urllib3 (1.22-1ubuntu0.18.04.2) ...\n",
            "Setting up python-chardet (3.0.4-1) ...\n",
            "Setting up nodejs (8.10.0~dfsg-2ubuntu0.4) ...\n",
            "update-alternatives: using /usr/bin/nodejs to provide /usr/bin/js (js) in auto mode\n",
            "Setting up python-rdflib (4.2.1-2) ...\n",
            "Setting up python-sparqlwrapper (1.7.6-2) ...\n",
            "Setting up python-rdflib-jsonld (0.4.0-3) ...\n",
            "Setting up python-cryptography (2.1.4-1ubuntu1.4) ...\n",
            "Setting up python-requests (2.18.4-2ubuntu0.1) ...\n",
            "Setting up python-cachecontrol (0.11.7-1) ...\n",
            "Setting up python-schema-salad (2.6.20171201034858-3) ...\n",
            "Setting up python-html5lib (0.999999999-1) ...\n",
            "Setting up samtools (1.7-1) ...\n",
            "Setting up python-openssl (17.5.0-1ubuntu1) ...\n",
            "Setting up cwltool (1.0.20180302231433-1) ...\n",
            "Processing triggers for libc-bin (2.27-3ubuntu1.3) ...\n",
            "/sbin/ldconfig.real: file /usr/local/lib/libnet.so is truncated\n",
            "\n",
            "/sbin/ldconfig.real: file /usr/local/lib/libjawt.so is truncated\n",
            "\n",
            "/sbin/ldconfig.real: file /usr/local/lib/libjli.so is truncated\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/python3.7/dist-packages/ideep4py/lib/libmkldnn.so.0 is not a symbolic link\n",
            "\n",
            "Processing triggers for man-db (2.8.3-2ubuntu0.1) ...\n"
          ]
        }
      ],
      "source": [
        "!apt-get install bowtie2 samtools"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 6,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "JgTq4lFjq93N",
        "outputId": "053fdc4c-db86-451f-b6bb-861bf26d1fc3"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\bdone\n",
            "Solving environment: - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\bdone\n",
            "\n",
            "## Package Plan ##\n",
            "\n",
            "  environment location: /usr/local\n",
            "\n",
            "  added / updated specs:\n",
            "    - bowtie2\n",
            "\n",
            "\n",
            "The following packages will be downloaded:\n",
            "\n",
            "    package                    |            build\n",
            "    ---------------------------|-----------------\n",
            "    bowtie2-2.3.5.1            |   py37he513fc3_0        12.0 MB  bioconda\n",
            "    perl-5.32.1                | 0_h7f98852_perl5        14.5 MB  conda-forge\n",
            "    tbb-2020.2                 |       h4bd325d_4         1.5 MB  conda-forge\n",
            "    ------------------------------------------------------------\n",
            "                                           Total:        27.9 MB\n",
            "\n",
            "The following NEW packages will be INSTALLED:\n",
            "\n",
            "  bowtie2            bioconda/linux-64::bowtie2-2.3.5.1-py37he513fc3_0\n",
            "  perl               conda-forge/linux-64::perl-5.32.1-0_h7f98852_perl5\n",
            "  tbb                conda-forge/linux-64::tbb-2020.2-h4bd325d_4\n",
            "\n",
            "\n",
            "\n",
            "Downloading and Extracting Packages\n",
            "perl-5.32.1          | 14.5 MB   | : 100% 1.0/1 [00:05<00:00,  5.35s/it]               \n",
            "bowtie2-2.3.5.1      | 12.0 MB   | : 100% 1.0/1 [00:03<00:00,  3.16s/it]               \n",
            "tbb-2020.2           | 1.5 MB    | : 100% 1.0/1 [00:00<00:00,  2.96it/s]\n",
            "Preparing transaction: - \b\bdone\n",
            "Verifying transaction: | \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\bdone\n",
            "Executing transaction: - \b\b\\ \b\b| \b\bdone\n"
          ]
        }
      ],
      "source": [
        "!conda install -c bioconda bowtie2"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "vcr5CjibHTgF"
      },
      "source": [
        "## Установка FastQC"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "62l1Q2ymbENA",
        "outputId": "2648cc5d-b4b5-4d83-e202-8f82b1473629"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "openjdk version \"1.8.0_312\"\n",
            "OpenJDK Runtime Environment (Zulu 8.58.0.13-CA-linux64) (build 1.8.0_312-b07)\n",
            "OpenJDK 64-Bit Server VM (Zulu 8.58.0.13-CA-linux64) (build 25.312-b07, mixed mode)\n"
          ]
        }
      ],
      "source": [
        "!java -version"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 1,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "8bpgwFgcbgRd",
        "outputId": "9b873787-5d0e-4175-a250-58ab84b8157f"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2022-03-10 20:19:55--  https://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.11.9.zip\n",
            "Resolving www.bioinformatics.babraham.ac.uk (www.bioinformatics.babraham.ac.uk)... 149.155.133.4\n",
            "Connecting to www.bioinformatics.babraham.ac.uk (www.bioinformatics.babraham.ac.uk)|149.155.133.4|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 10249221 (9.8M) [application/zip]\n",
            "Saving to: ‘fastqc_v0.11.9.zip’\n",
            "\n",
            "fastqc_v0.11.9.zip   95%[==================> ]   9.30M  62.2KB/s    in 2m 25s  \n",
            "\n",
            "2022-03-10 20:22:55 (65.7 KB/s) - Read error at byte 9751849/10249221 (Connection reset by peer). Retrying.\n",
            "\n",
            "--2022-03-10 20:22:56--  (try: 2)  https://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.11.9.zip\n",
            "Connecting to www.bioinformatics.babraham.ac.uk (www.bioinformatics.babraham.ac.uk)|149.155.133.4|:443... connected.\n",
            "HTTP request sent, awaiting response... 206 Partial Content\n",
            "Length: 10249221 (9.8M), 497372 (486K) remaining [application/zip]\n",
            "Saving to: ‘fastqc_v0.11.9.zip’\n",
            "\n",
            "fastqc_v0.11.9.zip  100%[+++++++++++++++++++>]   9.77M  82.7KB/s    in 5.9s    \n",
            "\n",
            "2022-03-10 20:23:03 (82.7 KB/s) - ‘fastqc_v0.11.9.zip’ saved [10249221/10249221]\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!wget https://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.11.9.zip"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 7,
      "metadata": {
        "id": "yTcTjOTJblcm",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "51f5f48c-79ce-464d-fbee-8664f6bb85dd"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Archive:  fastqc_v0.11.9.zip\n",
            "  inflating: FastQC/cisd-jhdf5.jar   \n",
            "   creating: FastQC/Configuration/\n",
            "  inflating: FastQC/Configuration/adapter_list.txt  \n",
            "  inflating: FastQC/Configuration/contaminant_list.txt  \n",
            "  inflating: FastQC/Configuration/limits.txt  \n",
            "  inflating: FastQC/fastqc           \n",
            "  inflating: FastQC/fastqc_icon.ico  \n",
            "   creating: FastQC/Help/\n",
            "   creating: FastQC/Help/1 Introduction/\n",
            "   creating: FastQC/Help/1 Introduction/.svn/\n",
            "  inflating: FastQC/Help/1 Introduction/.svn/entries  \n",
            "   creating: FastQC/Help/1 Introduction/.svn/props/\n",
            "   creating: FastQC/Help/1 Introduction/.svn/text-base/\n",
            "  inflating: FastQC/Help/1 Introduction/.svn/text-base/1.1 What is FastQC.html.svn-base  \n",
            "   creating: FastQC/Help/1 Introduction/.svn/tmp/\n",
            "   creating: FastQC/Help/1 Introduction/.svn/tmp/props/\n",
            "  inflating: FastQC/Help/1 Introduction/1.1 What is FastQC.html  \n",
            "   creating: FastQC/Help/2 Basic Operations/\n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/\n",
            "  inflating: FastQC/Help/2 Basic Operations/.svn/entries  \n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/props/\n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/text-base/\n",
            "  inflating: FastQC/Help/2 Basic Operations/.svn/text-base/2.1 Opening a sequence file.html.svn-base  \n",
            "  inflating: FastQC/Help/2 Basic Operations/.svn/text-base/2.2 Evaluating Results.html.svn-base  \n",
            "  inflating: FastQC/Help/2 Basic Operations/.svn/text-base/2.3 Saving a Report.html.svn-base  \n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/tmp/\n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/tmp/props/\n",
            "  inflating: FastQC/Help/2 Basic Operations/2.1 Opening a sequence file.html  \n",
            "  inflating: FastQC/Help/2 Basic Operations/2.2 Evaluating Results.html  \n",
            "  inflating: FastQC/Help/2 Basic Operations/2.3 Saving a Report.html  \n",
            "   creating: FastQC/Help/3 Analysis Modules/\n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/\n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/entries  \n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/prop-base/\n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/duplication_levels.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/prop-base/kmer_profiles.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_gc_content.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_n_content.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_quality.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_sequence_content.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_sequence_gc_content.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_sequence_quality.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_tile_quality.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/sequence_length_distribution.png.svn-base  \n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/props/\n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/text-base/\n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/1 Basic Statistics.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/10 Adapter Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/11 Kmer Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/12 Per Tile Sequence Quality.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/2 Per Base Sequence Quality.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/3 Per Sequence Quality Scores.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/4 Per Base Sequence Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/5 Per Sequence GC Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/6 Per Base N Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/7 Sequence Length Distribution.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/8 Duplicate Sequences.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/9 Overrepresented Sequences.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/duplication_levels.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/kmer_profiles.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_gc_content.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_n_content.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_quality.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_sequence_content.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_sequence_gc_content.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_sequence_quality.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_tile_quality.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/sequence_length_distribution.png.svn-base  \n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/tmp/\n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/tmp/props/\n",
            "  inflating: FastQC/Help/3 Analysis Modules/1 Basic Statistics.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/10 Adapter Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/11 Kmer Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/12 Per Tile Sequence Quality.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/2 Per Base Sequence Quality.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/3 Per Sequence Quality Scores.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/4 Per Base Sequence Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/5 Per Sequence GC Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/6 Per Base N Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/7 Sequence Length Distribution.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/8 Duplicate Sequences.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/9 Overrepresented Sequences.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/duplication_levels.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/kmer_profiles.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_base_gc_content.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_base_n_content.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_base_quality.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_base_sequence_content.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_sequence_gc_content.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_sequence_quality.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_tile_quality.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/sequence_length_distribution.png  \n",
            "  inflating: FastQC/INSTALL.txt      \n",
            "  inflating: FastQC/jbzip2-0.9.jar   \n",
            "  inflating: FastQC/LICENSE          \n",
            "  inflating: FastQC/LICENSE.txt      \n",
            "  inflating: FastQC/LICENSE_JHDF5.txt  \n",
            "   creating: FastQC/net/\n",
            "   creating: FastQC/net/sourceforge/\n",
            "   creating: FastQC/net/sourceforge/iharder/\n",
            "   creating: FastQC/net/sourceforge/iharder/base64/\n",
            "  inflating: FastQC/net/sourceforge/iharder/base64/Base64$1.class  \n",
            "  inflating: FastQC/net/sourceforge/iharder/base64/Base64$InputStream.class  \n",
            "  inflating: FastQC/net/sourceforge/iharder/base64/Base64$OutputStream.class  \n",
            "  inflating: FastQC/net/sourceforge/iharder/base64/Base64.class  \n",
            "   creating: FastQC/org/\n",
            "   creating: FastQC/org/apache/\n",
            "   creating: FastQC/org/apache/commons/\n",
            "   creating: FastQC/org/apache/commons/math3/\n",
            "   creating: FastQC/org/apache/commons/math3/analysis/\n",
            "   creating: FastQC/org/apache/commons/math3/analysis/solvers/\n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/AbstractUnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/AllowedSolution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BaseAbstractUnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BaseUnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BracketedUnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BrentSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/UnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/UnivariateSolverUtils.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/UnivariateFunction.class  \n",
            "   creating: FastQC/org/apache/commons/math3/distribution/\n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/AbstractIntegerDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/AbstractRealDistribution$1.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/AbstractRealDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/BetaDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/BinomialDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/CauchyDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/ChiSquaredDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/FDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/GammaDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/HypergeometricDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/IntegerDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/NormalDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/PascalDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/PoissonDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/RealDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/SaddlePointExpansion.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/TDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/WeibullDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/ZipfDistribution.class  \n",
            "   creating: FastQC/org/apache/commons/math3/exception/\n",
            "  inflating: FastQC/org/apache/commons/math3/exception/ConvergenceException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/DimensionMismatchException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathArithmeticException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathIllegalArgumentException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathIllegalNumberException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathIllegalStateException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathInternalError.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MaxCountExceededException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NoBracketingException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NotFiniteNumberException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NotPositiveException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NotStrictlyPositiveException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NullArgumentException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NumberIsTooLargeException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NumberIsTooSmallException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/OutOfRangeException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/TooManyEvaluationsException.class  \n",
            "   creating: FastQC/org/apache/commons/math3/exception/util/\n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/ArgUtils.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/ExceptionContext.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/ExceptionContextProvider.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/Localizable.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/LocalizedFormats.class  \n",
            "   creating: FastQC/org/apache/commons/math3/random/\n",
            "  inflating: FastQC/org/apache/commons/math3/random/AbstractWell.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/BitsStreamGenerator.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/RandomData.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/RandomDataImpl.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/RandomGenerator.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/Well19937c.class  \n",
            "   creating: FastQC/org/apache/commons/math3/special/\n",
            "  inflating: FastQC/org/apache/commons/math3/special/Beta$1.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/special/Beta.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/special/Erf.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/special/Gamma$1.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/special/Gamma.class  \n",
            "   creating: FastQC/org/apache/commons/math3/util/\n",
            "  inflating: FastQC/org/apache/commons/math3/util/ArithmeticUtils.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/ContinuedFraction.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/DoubleArray.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMath$ExpFracTable.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMath$ExpIntTable.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMath$lnMant.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMath.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMathCalc.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMathLiteralArrays.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/Incrementor$1.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/Incrementor$MaxCountExceededCallback.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/Incrementor.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/MathUtils.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/Precision.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/ResizableDoubleArray.class  \n",
            "  inflating: FastQC/README.md        \n",
            "  inflating: FastQC/README.txt       \n",
            "  inflating: FastQC/RELEASE_NOTES.txt  \n",
            "  inflating: FastQC/run_fastqc.bat   \n",
            "  inflating: FastQC/sam-1.103.jar    \n",
            "   creating: FastQC/Templates/\n",
            "  inflating: FastQC/Templates/fastqc2fo.xsl  \n",
            "  inflating: FastQC/Templates/header_template.html  \n",
            "   creating: FastQC/Templates/Icons/\n",
            " extracting: FastQC/Templates/Icons/error.png  \n",
            " extracting: FastQC/Templates/Icons/fastqc_icon.png  \n",
            " extracting: FastQC/Templates/Icons/tick.png  \n",
            " extracting: FastQC/Templates/Icons/warning.png  \n",
            "   creating: FastQC/uk/\n",
            "   creating: FastQC/uk/ac/\n",
            "   creating: FastQC/uk/ac/babraham/\n",
            "   creating: FastQC/uk/ac/babraham/FastQC/\n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Analysis/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/AnalysisListener.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/AnalysisQueue.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/AnalysisRunner.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/OfflineRunner.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Dialogs/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/AboutDialog$1.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/AboutDialog.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/FastQCTitlePanel$SmoothJLabel.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/FastQCTitlePanel.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/WelcomePanel.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCApplication$1.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCApplication.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCConfig.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCMenuBar.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/FileFilters/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/BAMFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/CasavaFastQFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/FastQFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/GobyFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/MappedBAMFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/SequenceFileFilter.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Graphs/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/BaseGroup.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/LineGraph.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/QualityBoxPlot.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/TileGraph.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Help/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpDialog.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpIndexRoot$FileSorter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpIndexRoot.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpPage.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpPageDisplay$HelpEditor.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpPageDisplay.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpSearchPanel.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Modules/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AbstractQCModule.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AdapterContent$Adapter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AdapterContent$ResultsTable.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AdapterContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/BasicStats$ResultsTable.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/BasicStats.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/DuplicationLevel.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Modules/GCModel/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/GCModel/GCModel.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/GCModel/GCModelValue.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/KmerContent$Kmer.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/KmerContent$ResultsTable.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/KmerContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/ModuleConfig.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/ModuleFactory.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/NContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/OverRepresentedSeqs$OverrepresentedSeq.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/OverRepresentedSeqs$ResultsTable.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/OverRepresentedSeqs.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerBaseQualityScores.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerBaseSequenceContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerSequenceGCContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerSequenceQualityScores.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerTileQualityScores.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/QCModule.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/SequenceLengthDistribution.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Report/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Report/HTMLReportArchive.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Report/stylesheet.txt  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Resources/\n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/error.png  \n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/fastqc_icon.png  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Resources/fastqc_icon.svg  \n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/fastqc_icon_100.png  \n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/tick.png  \n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/warning.png  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Results/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Results/ResultsPanel$ModuleRenderer.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Results/ResultsPanel.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Sequence/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/BAMFile.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/Contaminant.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/ContaminantHit.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/ContaminentFinder.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Fast5File.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/FastQFile.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Sequence/QualityEncoding/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/QualityEncoding/PhredEncoding.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Sequence.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFactory.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFile.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFileGroup.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFormatException.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Statistics/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Statistics/NormalDistribution.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Statistics/PearsonCorrelation.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Utilities/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/CasavaBasename.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/HotColdColourGradient.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/ImageToBase64.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/MultiMemberGZIPInputStream.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/NameFormatException.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/NanoporeBasename.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/QualityCount.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/RGB.class  \n"
          ]
        }
      ],
      "source": [
        "!unzip fastqc_v0.11.9.zip"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 8,
      "metadata": {
        "id": "S79LH0aXbvLx"
      },
      "outputs": [],
      "source": [
        "!chmod a+x FastQC/fastqc"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 9,
      "metadata": {
        "id": "PsLuZDt4b7jT",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "1202116c-8fef-4caa-a413-850b414ff616"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "            FastQC - A high throughput sequence QC analysis tool\n",
            "\n",
            "SYNOPSIS\n",
            "\n",
            "\tfastqc seqfile1 seqfile2 .. seqfileN\n",
            "\n",
            "    fastqc [-o output dir] [--(no)extract] [-f fastq|bam|sam] \n",
            "           [-c contaminant file] seqfile1 .. seqfileN\n",
            "\n",
            "DESCRIPTION\n",
            "\n",
            "    FastQC reads a set of sequence files and produces from each one a quality\n",
            "    control report consisting of a number of different modules, each one of \n",
            "    which will help to identify a different potential type of problem in your\n",
            "    data.\n",
            "    \n",
            "    If no files to process are specified on the command line then the program\n",
            "    will start as an interactive graphical application.  If files are provided\n",
            "    on the command line then the program will run with no user interaction\n",
            "    required.  In this mode it is suitable for inclusion into a standardised\n",
            "    analysis pipeline.\n",
            "    \n",
            "    The options for the program as as follows:\n",
            "    \n",
            "    -h --help       Print this help file and exit\n",
            "    \n",
            "    -v --version    Print the version of the program and exit\n",
            "    \n",
            "    -o --outdir     Create all output files in the specified output directory.\n",
            "                    Please note that this directory must exist as the program\n",
            "                    will not create it.  If this option is not set then the \n",
            "                    output file for each sequence file is created in the same\n",
            "                    directory as the sequence file which was processed.\n",
            "                    \n",
            "    --casava        Files come from raw casava output. Files in the same sample\n",
            "                    group (differing only by the group number) will be analysed\n",
            "                    as a set rather than individually. Sequences with the filter\n",
            "                    flag set in the header will be excluded from the analysis.\n",
            "                    Files must have the same names given to them by casava\n",
            "                    (including being gzipped and ending with .gz) otherwise they\n",
            "                    won't be grouped together correctly.\n",
            "                    \n",
            "    --nano          Files come from nanopore sequences and are in fast5 format. In\n",
            "                    this mode you can pass in directories to process and the program\n",
            "                    will take in all fast5 files within those directories and produce\n",
            "                    a single output file from the sequences found in all files.                    \n",
            "                    \n",
            "    --nofilter      If running with --casava then don't remove read flagged by\n",
            "                    casava as poor quality when performing the QC analysis.\n",
            "                   \n",
            "    --extract       If set then the zipped output file will be uncompressed in\n",
            "                    the same directory after it has been created.  By default\n",
            "                    this option will be set if fastqc is run in non-interactive\n",
            "                    mode.\n",
            "                    \n",
            "    -j --java       Provides the full path to the java binary you want to use to\n",
            "                    launch fastqc. If not supplied then java is assumed to be in\n",
            "                    your path.\n",
            "                   \n",
            "    --noextract     Do not uncompress the output file after creating it.  You\n",
            "                    should set this option if you do not wish to uncompress\n",
            "                    the output when running in non-interactive mode.\n",
            "                    \n",
            "    --nogroup       Disable grouping of bases for reads >50bp. All reports will\n",
            "                    show data for every base in the read.  WARNING: Using this\n",
            "                    option will cause fastqc to crash and burn if you use it on\n",
            "                    really long reads, and your plots may end up a ridiculous size.\n",
            "                    You have been warned!\n",
            "                    \n",
            "    --min_length    Sets an artificial lower limit on the length of the sequence\n",
            "                    to be shown in the report.  As long as you set this to a value\n",
            "                    greater or equal to your longest read length then this will be\n",
            "                    the sequence length used to create your read groups.  This can\n",
            "                    be useful for making directly comaparable statistics from \n",
            "                    datasets with somewhat variable read lengths.\n",
            "                    \n",
            "    -f --format     Bypasses the normal sequence file format detection and\n",
            "                    forces the program to use the specified format.  Valid\n",
            "                    formats are bam,sam,bam_mapped,sam_mapped and fastq\n",
            "                    \n",
            "    -t --threads    Specifies the number of files which can be processed\n",
            "                    simultaneously.  Each thread will be allocated 250MB of\n",
            "                    memory so you shouldn't run more threads than your\n",
            "                    available memory will cope with, and not more than\n",
            "                    6 threads on a 32 bit machine\n",
            "                  \n",
            "    -c              Specifies a non-default file which contains the list of\n",
            "    --contaminants  contaminants to screen overrepresented sequences against.\n",
            "                    The file must contain sets of named contaminants in the\n",
            "                    form name[tab]sequence.  Lines prefixed with a hash will\n",
            "                    be ignored.\n",
            "\n",
            "    -a              Specifies a non-default file which contains the list of\n",
            "    --adapters      adapter sequences which will be explicity searched against\n",
            "                    the library. The file must contain sets of named adapters\n",
            "                    in the form name[tab]sequence.  Lines prefixed with a hash\n",
            "                    will be ignored.\n",
            "                    \n",
            "    -l              Specifies a non-default file which contains a set of criteria\n",
            "    --limits        which will be used to determine the warn/error limits for the\n",
            "                    various modules.  This file can also be used to selectively \n",
            "                    remove some modules from the output all together.  The format\n",
            "                    needs to mirror the default limits.txt file found in the\n",
            "                    Configuration folder.\n",
            "                    \n",
            "   -k --kmers       Specifies the length of Kmer to look for in the Kmer content\n",
            "                    module. Specified Kmer length must be between 2 and 10. Default\n",
            "                    length is 7 if not specified.\n",
            "                    \n",
            "   -q --quiet       Supress all progress messages on stdout and only report errors.\n",
            "   \n",
            "   -d --dir         Selects a directory to be used for temporary files written when\n",
            "                    generating report images. Defaults to system temp directory if\n",
            "                    not specified.\n",
            "                    \n",
            "BUGS\n",
            "\n",
            "    Any bugs in fastqc should be reported either to simon.andrews@babraham.ac.uk\n",
            "    or in www.bioinformatics.babraham.ac.uk/bugzilla/\n",
            "                   \n",
            "    \n"
          ]
        }
      ],
      "source": [
        "!./FastQC/fastqc --help"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "10Fs3qrmHmJi"
      },
      "source": [
        "# Выравнивание"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "-z4GQiMfU1wf"
      },
      "source": [
        "## Скачиваем чтения\n",
        "\n",
        "Для примера чтения были взяты из эксперимента https://www.encodeproject.org/experiments/ENCSR000AKC/"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 10,
      "metadata": {
        "id": "lFWnMg0uU0sJ",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "8a3fa96b-9d5b-45b7-ea70-90f33165c619"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2022-03-10 20:29:11--  https://www.encodeproject.org/files/ENCFF019SLS/@@download/ENCFF019SLS.fastq.gz\n",
            "Resolving www.encodeproject.org (www.encodeproject.org)... 34.211.244.144\n",
            "Connecting to www.encodeproject.org (www.encodeproject.org)|34.211.244.144|:443... connected.\n",
            "HTTP request sent, awaiting response... 307 Temporary Redirect\n",
            "Location: https://encode-public.s3.amazonaws.com/2019/04/27/056e7433-bb81-4591-8271-57fdb881ba89/ENCFF019SLS.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF019SLS.fastq.gz&AWSAccessKeyId=ASIATGZNGCNXZWYIVGBT&Signature=yK5b%2FsC7LuTuM7ME7TvMdnfPhgw%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEIz%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FwEaCXVzLXdlc3QtMiJIMEYCIQDaY8r2KIMQ8ARjUrel4Z6jykxbuFMGVcD0w7VvEM1OiQIhAPKUaxlYciwOyweAaoEYFpKNivlgI1vuXAg8hxwiTzZmKoMECPX%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FwEQABoMMjIwNzQ4NzE0ODYzIgzBB7D2VncA0vaa8%2FQq1wPCjkTyUMOR5eVVmBJuyCcR0UYJcNEXpNK2e%2BJV9t9bUw6BryIvV7Yfn28SsUnpHlFQbvU7BSuPEkRQHuk3%2B%2BppyOMX%2BESx5EKkqOnRwBdSrQscAGJoxgkYSXgnkajWY6V4YZsHGY1%2FiufF259ZXCtApRqRuCLdxr7zC8069AAUKMT6fv5uqPb1C4tXsUrSBEwLvvyJ5Y1ttunrGwxih3fvx3b%2F1ZPqM4YSfL%2B25SqA0Wlz69PT7jwPOoCV2A94uSmROxTTf17kkqyMPEi3tat5nR7ZSdGsnueYAzFumnBWa7T2j51VqfQpjzUJDpY6ziJd2mMnPlILmCDXBA3C91aC%2FBJARmtjEksN7XXe6BdJVJlzfP7B8DRQdU6ck%2FISCgqLDWHFhZAA482Vz6hjcWaHIunrDu3qzl3eLY03dZZU9yx2V0kdE0ukl3W8RWjtOl%2B7Wa8WVeoIIrkEQDH%2FLqFNBXN8s3Wa7WDy%2F6D8w1jBdrlDdUB4D2kPtnvCZbR6J9TjdJ0DS0bc%2FrkxS3ktEd8p2SgSnD5ZoP4vE3OYQtCTkjqXjcVdf8BmUmpUTKhvrdGzLv9DU5BoiWn8w5DMz3KN%2Ber3BsqBWte8gKO54yg3w%2Bf3zRxlzekwobSpkQY6pAFPJozSjJNJo0xURHIS5Xe8bd13mNi8NaFYpe594jqzUQBGltmdkQeMBMy4TpcF%2BbXXW232jmHZHJAXhDggAaECiVOw7VzfuyErihSE1Q6R1g%2FGfcnLjYkC%2Be11vQWycDq81aku46koOKWZEtr3iomfG%2B8k9Zby3zDiuUMLJU5MN%2Fucr4OXqACbi7d6QtLORHMRQBSyZPNwVOBI1NFbC0xEocNFUA%3D%3D&Expires=1647073751 [following]\n",
            "--2022-03-10 20:29:11--  https://encode-public.s3.amazonaws.com/2019/04/27/056e7433-bb81-4591-8271-57fdb881ba89/ENCFF019SLS.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF019SLS.fastq.gz&AWSAccessKeyId=ASIATGZNGCNXZWYIVGBT&Signature=yK5b%2FsC7LuTuM7ME7TvMdnfPhgw%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEIz%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FwEaCXVzLXdlc3QtMiJIMEYCIQDaY8r2KIMQ8ARjUrel4Z6jykxbuFMGVcD0w7VvEM1OiQIhAPKUaxlYciwOyweAaoEYFpKNivlgI1vuXAg8hxwiTzZmKoMECPX%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FwEQABoMMjIwNzQ4NzE0ODYzIgzBB7D2VncA0vaa8%2FQq1wPCjkTyUMOR5eVVmBJuyCcR0UYJcNEXpNK2e%2BJV9t9bUw6BryIvV7Yfn28SsUnpHlFQbvU7BSuPEkRQHuk3%2B%2BppyOMX%2BESx5EKkqOnRwBdSrQscAGJoxgkYSXgnkajWY6V4YZsHGY1%2FiufF259ZXCtApRqRuCLdxr7zC8069AAUKMT6fv5uqPb1C4tXsUrSBEwLvvyJ5Y1ttunrGwxih3fvx3b%2F1ZPqM4YSfL%2B25SqA0Wlz69PT7jwPOoCV2A94uSmROxTTf17kkqyMPEi3tat5nR7ZSdGsnueYAzFumnBWa7T2j51VqfQpjzUJDpY6ziJd2mMnPlILmCDXBA3C91aC%2FBJARmtjEksN7XXe6BdJVJlzfP7B8DRQdU6ck%2FISCgqLDWHFhZAA482Vz6hjcWaHIunrDu3qzl3eLY03dZZU9yx2V0kdE0ukl3W8RWjtOl%2B7Wa8WVeoIIrkEQDH%2FLqFNBXN8s3Wa7WDy%2F6D8w1jBdrlDdUB4D2kPtnvCZbR6J9TjdJ0DS0bc%2FrkxS3ktEd8p2SgSnD5ZoP4vE3OYQtCTkjqXjcVdf8BmUmpUTKhvrdGzLv9DU5BoiWn8w5DMz3KN%2Ber3BsqBWte8gKO54yg3w%2Bf3zRxlzekwobSpkQY6pAFPJozSjJNJo0xURHIS5Xe8bd13mNi8NaFYpe594jqzUQBGltmdkQeMBMy4TpcF%2BbXXW232jmHZHJAXhDggAaECiVOw7VzfuyErihSE1Q6R1g%2FGfcnLjYkC%2Be11vQWycDq81aku46koOKWZEtr3iomfG%2B8k9Zby3zDiuUMLJU5MN%2Fucr4OXqACbi7d6QtLORHMRQBSyZPNwVOBI1NFbC0xEocNFUA%3D%3D&Expires=1647073751\n",
            "Resolving encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)... 52.218.138.67\n",
            "Connecting to encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)|52.218.138.67|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 253057716 (241M) [binary/octet-stream]\n",
            "Saving to: ‘ENCFF019SLS.fastq.gz’\n",
            "\n",
            "ENCFF019SLS.fastq.g 100%[===================>] 241.33M  32.8MB/s    in 8.2s    \n",
            "\n",
            "2022-03-10 20:29:20 (29.5 MB/s) - ‘ENCFF019SLS.fastq.gz’ saved [253057716/253057716]\n",
            "\n",
            "--2022-03-10 20:29:20--  https://www.encodeproject.org/files/ENCFF641KVN/@@download/ENCFF641KVN.fastq.gz\n",
            "Resolving www.encodeproject.org (www.encodeproject.org)... 34.211.244.144\n",
            "Connecting to www.encodeproject.org (www.encodeproject.org)|34.211.244.144|:443... connected.\n",
            "HTTP request sent, awaiting response... 307 Temporary Redirect\n",
            "Location: https://encode-public.s3.amazonaws.com/2020/01/31/e194e366-a721-4d16-8834-524a9e028732/ENCFF641KVN.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF641KVN.fastq.gz&AWSAccessKeyId=ASIATGZNGCNXZWYIVGBT&Signature=kcFPZ0fIImmhRFhUzZdPRv4Bv2U%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEIz%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FwEaCXVzLXdlc3QtMiJIMEYCIQDaY8r2KIMQ8ARjUrel4Z6jykxbuFMGVcD0w7VvEM1OiQIhAPKUaxlYciwOyweAaoEYFpKNivlgI1vuXAg8hxwiTzZmKoMECPX%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FwEQABoMMjIwNzQ4NzE0ODYzIgzBB7D2VncA0vaa8%2FQq1wPCjkTyUMOR5eVVmBJuyCcR0UYJcNEXpNK2e%2BJV9t9bUw6BryIvV7Yfn28SsUnpHlFQbvU7BSuPEkRQHuk3%2B%2BppyOMX%2BESx5EKkqOnRwBdSrQscAGJoxgkYSXgnkajWY6V4YZsHGY1%2FiufF259ZXCtApRqRuCLdxr7zC8069AAUKMT6fv5uqPb1C4tXsUrSBEwLvvyJ5Y1ttunrGwxih3fvx3b%2F1ZPqM4YSfL%2B25SqA0Wlz69PT7jwPOoCV2A94uSmROxTTf17kkqyMPEi3tat5nR7ZSdGsnueYAzFumnBWa7T2j51VqfQpjzUJDpY6ziJd2mMnPlILmCDXBA3C91aC%2FBJARmtjEksN7XXe6BdJVJlzfP7B8DRQdU6ck%2FISCgqLDWHFhZAA482Vz6hjcWaHIunrDu3qzl3eLY03dZZU9yx2V0kdE0ukl3W8RWjtOl%2B7Wa8WVeoIIrkEQDH%2FLqFNBXN8s3Wa7WDy%2F6D8w1jBdrlDdUB4D2kPtnvCZbR6J9TjdJ0DS0bc%2FrkxS3ktEd8p2SgSnD5ZoP4vE3OYQtCTkjqXjcVdf8BmUmpUTKhvrdGzLv9DU5BoiWn8w5DMz3KN%2Ber3BsqBWte8gKO54yg3w%2Bf3zRxlzekwobSpkQY6pAFPJozSjJNJo0xURHIS5Xe8bd13mNi8NaFYpe594jqzUQBGltmdkQeMBMy4TpcF%2BbXXW232jmHZHJAXhDggAaECiVOw7VzfuyErihSE1Q6R1g%2FGfcnLjYkC%2Be11vQWycDq81aku46koOKWZEtr3iomfG%2B8k9Zby3zDiuUMLJU5MN%2Fucr4OXqACbi7d6QtLORHMRQBSyZPNwVOBI1NFbC0xEocNFUA%3D%3D&Expires=1647073760 [following]\n",
            "--2022-03-10 20:29:20--  https://encode-public.s3.amazonaws.com/2020/01/31/e194e366-a721-4d16-8834-524a9e028732/ENCFF641KVN.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF641KVN.fastq.gz&AWSAccessKeyId=ASIATGZNGCNXZWYIVGBT&Signature=kcFPZ0fIImmhRFhUzZdPRv4Bv2U%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEIz%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FwEaCXVzLXdlc3QtMiJIMEYCIQDaY8r2KIMQ8ARjUrel4Z6jykxbuFMGVcD0w7VvEM1OiQIhAPKUaxlYciwOyweAaoEYFpKNivlgI1vuXAg8hxwiTzZmKoMECPX%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FwEQABoMMjIwNzQ4NzE0ODYzIgzBB7D2VncA0vaa8%2FQq1wPCjkTyUMOR5eVVmBJuyCcR0UYJcNEXpNK2e%2BJV9t9bUw6BryIvV7Yfn28SsUnpHlFQbvU7BSuPEkRQHuk3%2B%2BppyOMX%2BESx5EKkqOnRwBdSrQscAGJoxgkYSXgnkajWY6V4YZsHGY1%2FiufF259ZXCtApRqRuCLdxr7zC8069AAUKMT6fv5uqPb1C4tXsUrSBEwLvvyJ5Y1ttunrGwxih3fvx3b%2F1ZPqM4YSfL%2B25SqA0Wlz69PT7jwPOoCV2A94uSmROxTTf17kkqyMPEi3tat5nR7ZSdGsnueYAzFumnBWa7T2j51VqfQpjzUJDpY6ziJd2mMnPlILmCDXBA3C91aC%2FBJARmtjEksN7XXe6BdJVJlzfP7B8DRQdU6ck%2FISCgqLDWHFhZAA482Vz6hjcWaHIunrDu3qzl3eLY03dZZU9yx2V0kdE0ukl3W8RWjtOl%2B7Wa8WVeoIIrkEQDH%2FLqFNBXN8s3Wa7WDy%2F6D8w1jBdrlDdUB4D2kPtnvCZbR6J9TjdJ0DS0bc%2FrkxS3ktEd8p2SgSnD5ZoP4vE3OYQtCTkjqXjcVdf8BmUmpUTKhvrdGzLv9DU5BoiWn8w5DMz3KN%2Ber3BsqBWte8gKO54yg3w%2Bf3zRxlzekwobSpkQY6pAFPJozSjJNJo0xURHIS5Xe8bd13mNi8NaFYpe594jqzUQBGltmdkQeMBMy4TpcF%2BbXXW232jmHZHJAXhDggAaECiVOw7VzfuyErihSE1Q6R1g%2FGfcnLjYkC%2Be11vQWycDq81aku46koOKWZEtr3iomfG%2B8k9Zby3zDiuUMLJU5MN%2Fucr4OXqACbi7d6QtLORHMRQBSyZPNwVOBI1NFbC0xEocNFUA%3D%3D&Expires=1647073760\n",
            "Resolving encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)... 52.218.193.218\n",
            "Connecting to encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)|52.218.193.218|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 349798886 (334M) [binary/octet-stream]\n",
            "Saving to: ‘ENCFF641KVN.fastq.gz’\n",
            "\n",
            "ENCFF641KVN.fastq.g 100%[===================>] 333.59M  33.7MB/s    in 11s     \n",
            "\n",
            "2022-03-10 20:29:32 (31.1 MB/s) - ‘ENCFF641KVN.fastq.gz’ saved [349798886/349798886]\n",
            "\n",
            "--2022-03-10 20:29:32--  https://www.encodeproject.org/files/ENCFF411GHV/@@download/ENCFF411GHV.fastq.gz\n",
            "Resolving www.encodeproject.org (www.encodeproject.org)... 34.211.244.144\n",
            "Connecting to www.encodeproject.org (www.encodeproject.org)|34.211.244.144|:443... connected.\n",
            "HTTP request sent, awaiting response... 307 Temporary Redirect\n",
            "Location: https://encode-public.s3.amazonaws.com/2019/04/27/001e0172-9af3-45f7-bf9a-5506c9432ae7/ENCFF411GHV.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF411GHV.fastq.gz&AWSAccessKeyId=ASIATGZNGCNXZWYIVGBT&Signature=DXZquZ8AUt5uFLmuURg7FnB0AUc%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEIz%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FwEaCXVzLXdlc3QtMiJIMEYCIQDaY8r2KIMQ8ARjUrel4Z6jykxbuFMGVcD0w7VvEM1OiQIhAPKUaxlYciwOyweAaoEYFpKNivlgI1vuXAg8hxwiTzZmKoMECPX%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FwEQABoMMjIwNzQ4NzE0ODYzIgzBB7D2VncA0vaa8%2FQq1wPCjkTyUMOR5eVVmBJuyCcR0UYJcNEXpNK2e%2BJV9t9bUw6BryIvV7Yfn28SsUnpHlFQbvU7BSuPEkRQHuk3%2B%2BppyOMX%2BESx5EKkqOnRwBdSrQscAGJoxgkYSXgnkajWY6V4YZsHGY1%2FiufF259ZXCtApRqRuCLdxr7zC8069AAUKMT6fv5uqPb1C4tXsUrSBEwLvvyJ5Y1ttunrGwxih3fvx3b%2F1ZPqM4YSfL%2B25SqA0Wlz69PT7jwPOoCV2A94uSmROxTTf17kkqyMPEi3tat5nR7ZSdGsnueYAzFumnBWa7T2j51VqfQpjzUJDpY6ziJd2mMnPlILmCDXBA3C91aC%2FBJARmtjEksN7XXe6BdJVJlzfP7B8DRQdU6ck%2FISCgqLDWHFhZAA482Vz6hjcWaHIunrDu3qzl3eLY03dZZU9yx2V0kdE0ukl3W8RWjtOl%2B7Wa8WVeoIIrkEQDH%2FLqFNBXN8s3Wa7WDy%2F6D8w1jBdrlDdUB4D2kPtnvCZbR6J9TjdJ0DS0bc%2FrkxS3ktEd8p2SgSnD5ZoP4vE3OYQtCTkjqXjcVdf8BmUmpUTKhvrdGzLv9DU5BoiWn8w5DMz3KN%2Ber3BsqBWte8gKO54yg3w%2Bf3zRxlzekwobSpkQY6pAFPJozSjJNJo0xURHIS5Xe8bd13mNi8NaFYpe594jqzUQBGltmdkQeMBMy4TpcF%2BbXXW232jmHZHJAXhDggAaECiVOw7VzfuyErihSE1Q6R1g%2FGfcnLjYkC%2Be11vQWycDq81aku46koOKWZEtr3iomfG%2B8k9Zby3zDiuUMLJU5MN%2Fucr4OXqACbi7d6QtLORHMRQBSyZPNwVOBI1NFbC0xEocNFUA%3D%3D&Expires=1647073772 [following]\n",
            "--2022-03-10 20:29:32--  https://encode-public.s3.amazonaws.com/2019/04/27/001e0172-9af3-45f7-bf9a-5506c9432ae7/ENCFF411GHV.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF411GHV.fastq.gz&AWSAccessKeyId=ASIATGZNGCNXZWYIVGBT&Signature=DXZquZ8AUt5uFLmuURg7FnB0AUc%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEIz%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FwEaCXVzLXdlc3QtMiJIMEYCIQDaY8r2KIMQ8ARjUrel4Z6jykxbuFMGVcD0w7VvEM1OiQIhAPKUaxlYciwOyweAaoEYFpKNivlgI1vuXAg8hxwiTzZmKoMECPX%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FwEQABoMMjIwNzQ4NzE0ODYzIgzBB7D2VncA0vaa8%2FQq1wPCjkTyUMOR5eVVmBJuyCcR0UYJcNEXpNK2e%2BJV9t9bUw6BryIvV7Yfn28SsUnpHlFQbvU7BSuPEkRQHuk3%2B%2BppyOMX%2BESx5EKkqOnRwBdSrQscAGJoxgkYSXgnkajWY6V4YZsHGY1%2FiufF259ZXCtApRqRuCLdxr7zC8069AAUKMT6fv5uqPb1C4tXsUrSBEwLvvyJ5Y1ttunrGwxih3fvx3b%2F1ZPqM4YSfL%2B25SqA0Wlz69PT7jwPOoCV2A94uSmROxTTf17kkqyMPEi3tat5nR7ZSdGsnueYAzFumnBWa7T2j51VqfQpjzUJDpY6ziJd2mMnPlILmCDXBA3C91aC%2FBJARmtjEksN7XXe6BdJVJlzfP7B8DRQdU6ck%2FISCgqLDWHFhZAA482Vz6hjcWaHIunrDu3qzl3eLY03dZZU9yx2V0kdE0ukl3W8RWjtOl%2B7Wa8WVeoIIrkEQDH%2FLqFNBXN8s3Wa7WDy%2F6D8w1jBdrlDdUB4D2kPtnvCZbR6J9TjdJ0DS0bc%2FrkxS3ktEd8p2SgSnD5ZoP4vE3OYQtCTkjqXjcVdf8BmUmpUTKhvrdGzLv9DU5BoiWn8w5DMz3KN%2Ber3BsqBWte8gKO54yg3w%2Bf3zRxlzekwobSpkQY6pAFPJozSjJNJo0xURHIS5Xe8bd13mNi8NaFYpe594jqzUQBGltmdkQeMBMy4TpcF%2BbXXW232jmHZHJAXhDggAaECiVOw7VzfuyErihSE1Q6R1g%2FGfcnLjYkC%2Be11vQWycDq81aku46koOKWZEtr3iomfG%2B8k9Zby3zDiuUMLJU5MN%2Fucr4OXqACbi7d6QtLORHMRQBSyZPNwVOBI1NFbC0xEocNFUA%3D%3D&Expires=1647073772\n",
            "Resolving encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)... 52.218.169.99\n",
            "Connecting to encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)|52.218.169.99|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 106331223 (101M) [binary/octet-stream]\n",
            "Saving to: ‘ENCFF411GHV.fastq.gz’\n",
            "\n",
            "ENCFF411GHV.fastq.g 100%[===================>] 101.41M  32.0MB/s    in 3.5s    \n",
            "\n",
            "2022-03-10 20:29:36 (29.2 MB/s) - ‘ENCFF411GHV.fastq.gz’ saved [106331223/106331223]\n",
            "\n"
          ]
        }
      ],
      "source": [
        "# ChIP-seq на гистоновой метке\n",
        "!wget https://www.encodeproject.org/files/ENCFF019SLS/@@download/ENCFF019SLS.fastq.gz\n",
        "!wget https://www.encodeproject.org/files/ENCFF641KVN/@@download/ENCFF641KVN.fastq.gz\n",
        "# ChIP-seq контроль\n",
        "!wget https://www.encodeproject.org/files/ENCFF411GHV/@@download/ENCFF411GHV.fastq.gz\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 11,
      "metadata": {
        "id": "v73fH7WiVDu9"
      },
      "outputs": [],
      "source": [
        "# распаковка ахрива\n",
        "!gzip -d /content/ENCFF641KVN.fastq.gz /content/ENCFF411GHV.fastq.gz /content/ENCFF019SLS.fastq.gz"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "6sPcIefNHx-L"
      },
      "source": [
        "## FastQC (или multiQC может быть использован в работе)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 12,
      "metadata": {
        "id": "5XM6jBcuGjMV",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "8538df60-69f2-4f89-ea62-792200cd7a90"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Started analysis of ENCFF641KVN.fastq\n",
            "Approx 5% complete for ENCFF641KVN.fastq\n",
            "Approx 10% complete for ENCFF641KVN.fastq\n",
            "Approx 15% complete for ENCFF641KVN.fastq\n",
            "Approx 20% complete for ENCFF641KVN.fastq\n",
            "Approx 25% complete for ENCFF641KVN.fastq\n",
            "Approx 30% complete for ENCFF641KVN.fastq\n",
            "Approx 35% complete for ENCFF641KVN.fastq\n",
            "Approx 40% complete for ENCFF641KVN.fastq\n",
            "Approx 45% complete for ENCFF641KVN.fastq\n",
            "Approx 50% complete for ENCFF641KVN.fastq\n",
            "Approx 55% complete for ENCFF641KVN.fastq\n",
            "Approx 60% complete for ENCFF641KVN.fastq\n",
            "Approx 65% complete for ENCFF641KVN.fastq\n",
            "Approx 70% complete for ENCFF641KVN.fastq\n",
            "Approx 75% complete for ENCFF641KVN.fastq\n",
            "Approx 80% complete for ENCFF641KVN.fastq\n",
            "Approx 85% complete for ENCFF641KVN.fastq\n",
            "Approx 90% complete for ENCFF641KVN.fastq\n",
            "Approx 95% complete for ENCFF641KVN.fastq\n",
            "Analysis complete for ENCFF641KVN.fastq\n",
            "Started analysis of ENCFF411GHV.fastq\n",
            "Approx 5% complete for ENCFF411GHV.fastq\n",
            "Approx 10% complete for ENCFF411GHV.fastq\n",
            "Approx 15% complete for ENCFF411GHV.fastq\n",
            "Approx 20% complete for ENCFF411GHV.fastq\n",
            "Approx 25% complete for ENCFF411GHV.fastq\n",
            "Approx 30% complete for ENCFF411GHV.fastq\n",
            "Approx 35% complete for ENCFF411GHV.fastq\n",
            "Approx 40% complete for ENCFF411GHV.fastq\n",
            "Approx 45% complete for ENCFF411GHV.fastq\n",
            "Approx 50% complete for ENCFF411GHV.fastq\n",
            "Approx 55% complete for ENCFF411GHV.fastq\n",
            "Approx 60% complete for ENCFF411GHV.fastq\n",
            "Approx 65% complete for ENCFF411GHV.fastq\n",
            "Approx 70% complete for ENCFF411GHV.fastq\n",
            "Approx 75% complete for ENCFF411GHV.fastq\n",
            "Approx 80% complete for ENCFF411GHV.fastq\n",
            "Approx 85% complete for ENCFF411GHV.fastq\n",
            "Approx 90% complete for ENCFF411GHV.fastq\n",
            "Approx 95% complete for ENCFF411GHV.fastq\n",
            "Analysis complete for ENCFF411GHV.fastq\n",
            "Started analysis of ENCFF019SLS.fastq\n",
            "Approx 5% complete for ENCFF019SLS.fastq\n",
            "Approx 10% complete for ENCFF019SLS.fastq\n",
            "Approx 15% complete for ENCFF019SLS.fastq\n",
            "Approx 20% complete for ENCFF019SLS.fastq\n",
            "Approx 25% complete for ENCFF019SLS.fastq\n",
            "Approx 30% complete for ENCFF019SLS.fastq\n",
            "Approx 35% complete for ENCFF019SLS.fastq\n",
            "Approx 40% complete for ENCFF019SLS.fastq\n",
            "Approx 45% complete for ENCFF019SLS.fastq\n",
            "Approx 50% complete for ENCFF019SLS.fastq\n",
            "Approx 55% complete for ENCFF019SLS.fastq\n",
            "Approx 60% complete for ENCFF019SLS.fastq\n",
            "Approx 65% complete for ENCFF019SLS.fastq\n",
            "Approx 70% complete for ENCFF019SLS.fastq\n",
            "Approx 75% complete for ENCFF019SLS.fastq\n",
            "Approx 80% complete for ENCFF019SLS.fastq\n",
            "Approx 85% complete for ENCFF019SLS.fastq\n",
            "Approx 90% complete for ENCFF019SLS.fastq\n",
            "Approx 95% complete for ENCFF019SLS.fastq\n",
            "Analysis complete for ENCFF019SLS.fastq\n"
          ]
        }
      ],
      "source": [
        "!./FastQC/fastqc /content/ENCFF641KVN.fastq\n",
        "!./FastQC/fastqc /content/ENCFF411GHV.fastq\n",
        "!./FastQC/fastqc /content/ENCFF019SLS.fastq"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "W9QHWBLlpMjQ"
      },
      "source": [
        "## Выравнивание на хромосому\n",
        "\n",
        "Следует выбрать одну хромосому, потому что ресурсы google colab ограничены. В случае проблем, таких как Segmentation fault, error 139, следует выбрать хромосому меньшего размера.\n",
        "\n",
        "Последовательность нуклеотидов хромосомы можно скачать по адресу https://hgdownload.soe.ucsc.edu/goldenPath/hg38/chromosomes/"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 13,
      "metadata": {
        "id": "N4JIGZOsGiY2",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "c3e6fee0-3788-4cdf-d008-7e534863a5e3"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2022-03-10 20:32:04--  https://hgdownload.soe.ucsc.edu/goldenPath/hg38/chromosomes/chr18.fa.gz\n",
            "Resolving hgdownload.soe.ucsc.edu (hgdownload.soe.ucsc.edu)... 128.114.119.163\n",
            "Connecting to hgdownload.soe.ucsc.edu (hgdownload.soe.ucsc.edu)|128.114.119.163|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 25154367 (24M) [application/x-gzip]\n",
            "Saving to: ‘chr18.fa.gz’\n",
            "\n",
            "chr18.fa.gz         100%[===================>]  23.99M  18.9MB/s    in 1.3s    \n",
            "\n",
            "2022-03-10 20:32:06 (18.9 MB/s) - ‘chr18.fa.gz’ saved [25154367/25154367]\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!wget https://hgdownload.soe.ucsc.edu/goldenPath/hg38/chromosomes/chr18.fa.gz\n",
        "!gzip -d chr18.fa.gz"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 14,
      "metadata": {
        "id": "bhraQ6NdGiPN",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "76b472a3-adcb-4fb7-a01a-b752f818f132"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Settings:\n",
            "  Output files: \"chromosome_index.*.bt2\"\n",
            "  Line rate: 6 (line is 64 bytes)\n",
            "  Lines per side: 1 (side is 64 bytes)\n",
            "  Offset rate: 4 (one in 16)\n",
            "  FTable chars: 10\n",
            "  Strings: unpacked\n",
            "  Max bucket size: default\n",
            "  Max bucket size, sqrt multiplier: default\n",
            "  Max bucket size, len divisor: 4\n",
            "  Difference-cover sample period: 1024\n",
            "  Endianness: little\n",
            "  Actual local endianness: little\n",
            "  Sanity checking: disabled\n",
            "  Assertions: disabled\n",
            "  Random seed: 0\n",
            "  Sizeofs: void*:8, int:4, long:8, size_t:8\n",
            "Input files DNA, FASTA:\n",
            "  chr18.fa\n",
            "Building a SMALL index\n",
            "Reading reference sizes\n",
            "  Time reading reference sizes: 00:00:01\n",
            "Calculating joined length\n",
            "Writing header\n",
            "Reserving space for joined string\n",
            "Joining reference sequences\n",
            "  Time to join reference sequences: 00:00:00\n",
            "bmax according to bmaxDivN setting: 20022401\n",
            "Using parameters --bmax 15016801 --dcv 1024\n",
            "  Doing ahead-of-time memory usage test\n",
            "  Passed!  Constructing with these parameters: --bmax 15016801 --dcv 1024\n",
            "Constructing suffix-array element generator\n",
            "Building DifferenceCoverSample\n",
            "  Building sPrime\n",
            "  Building sPrimeOrder\n",
            "  V-Sorting samples\n",
            "  V-Sorting samples time: 00:00:02\n",
            "  Allocating rank array\n",
            "  Ranking v-sort output\n",
            "  Ranking v-sort output time: 00:00:00\n",
            "  Invoking Larsson-Sadakane on ranks\n",
            "  Invoking Larsson-Sadakane on ranks time: 00:00:01\n",
            "  Sanity-checking and returning\n",
            "Building samples\n",
            "Reserving space for 12 sample suffixes\n",
            "Generating random suffixes\n",
            "QSorting 12 sample offsets, eliminating duplicates\n",
            "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
            "Multikey QSorting 12 samples\n",
            "  (Using difference cover)\n",
            "  Multikey QSorting samples time: 00:00:00\n",
            "Calculating bucket sizes\n",
            "Splitting and merging\n",
            "  Splitting and merging time: 00:00:00\n",
            "Avg bucket size: 8.00896e+07 (target: 15016800)\n",
            "Converting suffix-array elements to index image\n",
            "Allocating ftab, absorbFtab\n",
            "Entering Ebwt loop\n",
            "Getting block 1 of 1\n",
            "  No samples; assembling all-inclusive block\n",
            "  Sorting block of length 80089605 for bucket 1\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:47\n",
            "Returning block of 80089606 for bucket 1\n",
            "Exited Ebwt loop\n",
            "fchr[A]: 0\n",
            "fchr[C]: 24050680\n",
            "fchr[G]: 39845135\n",
            "fchr[T]: 55906786\n",
            "fchr[$]: 80089605\n",
            "Exiting Ebwt::buildToDisk()\n",
            "Returning from initFromVector\n",
            "Wrote 30891755 bytes to primary EBWT file: chromosome_index.1.bt2\n",
            "Wrote 20022408 bytes to secondary EBWT file: chromosome_index.2.bt2\n",
            "Re-opening _in1 and _in2 as input streams\n",
            "Returning from Ebwt constructor\n",
            "Headers:\n",
            "    len: 80089605\n",
            "    bwtLen: 80089606\n",
            "    sz: 20022402\n",
            "    bwtSz: 20022402\n",
            "    lineRate: 6\n",
            "    offRate: 4\n",
            "    offMask: 0xfffffff0\n",
            "    ftabChars: 10\n",
            "    eftabLen: 20\n",
            "    eftabSz: 80\n",
            "    ftabLen: 1048577\n",
            "    ftabSz: 4194308\n",
            "    offsLen: 5005601\n",
            "    offsSz: 20022404\n",
            "    lineSz: 64\n",
            "    sideSz: 64\n",
            "    sideBwtSz: 48\n",
            "    sideBwtLen: 192\n",
            "    numSides: 417134\n",
            "    numLines: 417134\n",
            "    ebwtTotLen: 26696576\n",
            "    ebwtTotSz: 26696576\n",
            "    color: 0\n",
            "    reverse: 0\n",
            "Total time for call to driver() for forward index: 00:01:04\n",
            "Reading reference sizes\n",
            "  Time reading reference sizes: 00:00:01\n",
            "Calculating joined length\n",
            "Writing header\n",
            "Reserving space for joined string\n",
            "Joining reference sequences\n",
            "  Time to join reference sequences: 00:00:00\n",
            "  Time to reverse reference sequence: 00:00:00\n",
            "bmax according to bmaxDivN setting: 20022401\n",
            "Using parameters --bmax 15016801 --dcv 1024\n",
            "  Doing ahead-of-time memory usage test\n",
            "  Passed!  Constructing with these parameters: --bmax 15016801 --dcv 1024\n",
            "Constructing suffix-array element generator\n",
            "Building DifferenceCoverSample\n",
            "  Building sPrime\n",
            "  Building sPrimeOrder\n",
            "  V-Sorting samples\n",
            "  V-Sorting samples time: 00:00:02\n",
            "  Allocating rank array\n",
            "  Ranking v-sort output\n",
            "  Ranking v-sort output time: 00:00:01\n",
            "  Invoking Larsson-Sadakane on ranks\n",
            "  Invoking Larsson-Sadakane on ranks time: 00:00:01\n",
            "  Sanity-checking and returning\n",
            "Building samples\n",
            "Reserving space for 12 sample suffixes\n",
            "Generating random suffixes\n",
            "QSorting 12 sample offsets, eliminating duplicates\n",
            "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
            "Multikey QSorting 12 samples\n",
            "  (Using difference cover)\n",
            "  Multikey QSorting samples time: 00:00:00\n",
            "Calculating bucket sizes\n",
            "Splitting and merging\n",
            "  Splitting and merging time: 00:00:00\n",
            "Avg bucket size: 8.00896e+07 (target: 15016800)\n",
            "Converting suffix-array elements to index image\n",
            "Allocating ftab, absorbFtab\n",
            "Entering Ebwt loop\n",
            "Getting block 1 of 1\n",
            "  No samples; assembling all-inclusive block\n",
            "  Sorting block of length 80089605 for bucket 1\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:46\n",
            "Returning block of 80089606 for bucket 1\n",
            "Exited Ebwt loop\n",
            "fchr[A]: 0\n",
            "fchr[C]: 24050680\n",
            "fchr[G]: 39845135\n",
            "fchr[T]: 55906786\n",
            "fchr[$]: 80089605\n",
            "Exiting Ebwt::buildToDisk()\n",
            "Returning from initFromVector\n",
            "Wrote 30891755 bytes to primary EBWT file: chromosome_index.rev.1.bt2\n",
            "Wrote 20022408 bytes to secondary EBWT file: chromosome_index.rev.2.bt2\n",
            "Re-opening _in1 and _in2 as input streams\n",
            "Returning from Ebwt constructor\n",
            "Headers:\n",
            "    len: 80089605\n",
            "    bwtLen: 80089606\n",
            "    sz: 20022402\n",
            "    bwtSz: 20022402\n",
            "    lineRate: 6\n",
            "    offRate: 4\n",
            "    offMask: 0xfffffff0\n",
            "    ftabChars: 10\n",
            "    eftabLen: 20\n",
            "    eftabSz: 80\n",
            "    ftabLen: 1048577\n",
            "    ftabSz: 4194308\n",
            "    offsLen: 5005601\n",
            "    offsSz: 20022404\n",
            "    lineSz: 64\n",
            "    sideSz: 64\n",
            "    sideBwtSz: 48\n",
            "    sideBwtLen: 192\n",
            "    numSides: 417134\n",
            "    numLines: 417134\n",
            "    ebwtTotLen: 26696576\n",
            "    ebwtTotSz: 26696576\n",
            "    color: 0\n",
            "    reverse: 1\n",
            "Total time for backward call to driver() for mirror index: 00:01:03\n",
            "CPU times: user 1.01 s, sys: 156 ms, total: 1.17 s\n",
            "Wall time: 2min 6s\n"
          ]
        }
      ],
      "source": [
        "%%time\n",
        "!bowtie2-build chr18.fa chromosome_index"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 15,
      "metadata": {
        "id": "E_usI8XUrQnR"
      },
      "outputs": [],
      "source": [
        "!mkdir bowtie2_res"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 16,
      "metadata": {
        "id": "3t80UATHqFe7",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "ddcb932a-3c1a-462b-f661-a91ff288518a"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "4697829 reads; of these:\n",
            "  4697829 (100.00%) were unpaired; of these:\n",
            "    4478193 (95.32%) aligned 0 times\n",
            "    100272 (2.13%) aligned exactly 1 time\n",
            "    119364 (2.54%) aligned >1 times\n",
            "4.68% overall alignment rate\n",
            "CPU times: user 1.46 s, sys: 196 ms, total: 1.65 s\n",
            "Wall time: 3min 13s\n"
          ]
        }
      ],
      "source": [
        "%%time\n",
        "!bowtie2  -p 2 \\\n",
        "          -x chromosome_index \\\n",
        "          -U ENCFF019SLS.fastq \\\n",
        "          -S bowtie2_res/res_SLS.sam"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 17,
      "metadata": {
        "id": "so1jTdGHoEir",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "9288f572-dd19-46b9-cd89-1df788a6fbcc"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "6467900 reads; of these:\n",
            "  6467900 (100.00%) were unpaired; of these:\n",
            "    6181873 (95.58%) aligned 0 times\n",
            "    140146 (2.17%) aligned exactly 1 time\n",
            "    145881 (2.26%) aligned >1 times\n",
            "4.42% overall alignment rate\n",
            "CPU times: user 2.03 s, sys: 273 ms, total: 2.3 s\n",
            "Wall time: 4min 19s\n"
          ]
        }
      ],
      "source": [
        "%%time\n",
        "!bowtie2  -p 2 \\\n",
        "          -x chromosome_index \\\n",
        "          -U ENCFF641KVN.fastq \\\n",
        "          -S bowtie2_res/res_KVN.sam"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 18,
      "metadata": {
        "id": "zl05RSsIApYS",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "b7222b78-96b6-490e-a145-0656a2198296"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "2349952 reads; of these:\n",
            "  2349952 (100.00%) were unpaired; of these:\n",
            "    2106816 (89.65%) aligned 0 times\n",
            "    80393 (3.42%) aligned exactly 1 time\n",
            "    162743 (6.93%) aligned >1 times\n",
            "10.35% overall alignment rate\n",
            "CPU times: user 1.05 s, sys: 137 ms, total: 1.18 s\n",
            "Wall time: 2min 14s\n"
          ]
        }
      ],
      "source": [
        "%%time\n",
        "!bowtie2  -p 2 \\\n",
        "          -x chromosome_index \\\n",
        "          -U ENCFF411GHV.fastq \\\n",
        "          -S bowtie2_res/res_GHV.sam"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "ysoDrnSCrei0"
      },
      "source": [
        "Проанализируйте выдачу bowtie. Почему процент выравниваний получился именно таким?\n",
        "\n",
        "Имеет смысл для дальнейшего анализа отобрать уникально картированные риды."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 19,
      "metadata": {
        "id": "G-dZF2Z9CR24",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 160
        },
        "outputId": "0daa3fe3-7195-4b54-da1a-238abf0e5121"
      },
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/html": [
              "\n",
              "  <div id=\"df-99933817-6409-4bc2-a013-b958d2078fe7\">\n",
              "    <div class=\"colab-df-container\">\n",
              "      <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>Всего ридов</th>\n",
              "      <th>Выровнилось уникально</th>\n",
              "      <th>Выровнилось неуникально</th>\n",
              "      <th>Не выровнилось</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>ENCFF019SLS</th>\n",
              "      <td>4697829</td>\n",
              "      <td>100272</td>\n",
              "      <td>119364</td>\n",
              "      <td>4478193</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>ENCFF641KVN</th>\n",
              "      <td>6467900</td>\n",
              "      <td>140146</td>\n",
              "      <td>145881</td>\n",
              "      <td>6181873</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>ENCFF411GHV</th>\n",
              "      <td>2349952</td>\n",
              "      <td>80393</td>\n",
              "      <td>162743</td>\n",
              "      <td>2106816</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "      <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-99933817-6409-4bc2-a013-b958d2078fe7')\"\n",
              "              title=\"Convert this dataframe to an interactive table.\"\n",
              "              style=\"display:none;\">\n",
              "        \n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "       width=\"24px\">\n",
              "    <path d=\"M0 0h24v24H0V0z\" fill=\"none\"/>\n",
              "    <path d=\"M18.56 5.44l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94zm-11 1L8.5 8.5l.94-2.06 2.06-.94-2.06-.94L8.5 2.5l-.94 2.06-2.06.94zm10 10l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94z\"/><path d=\"M17.41 7.96l-1.37-1.37c-.4-.4-.92-.59-1.43-.59-.52 0-1.04.2-1.43.59L10.3 9.45l-7.72 7.72c-.78.78-.78 2.05 0 2.83L4 21.41c.39.39.9.59 1.41.59.51 0 1.02-.2 1.41-.59l7.78-7.78 2.81-2.81c.8-.78.8-2.07 0-2.86zM5.41 20L4 18.59l7.72-7.72 1.47 1.35L5.41 20z\"/>\n",
              "  </svg>\n",
              "      </button>\n",
              "      \n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      flex-wrap:wrap;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "      <script>\n",
              "        const buttonEl =\n",
              "          document.querySelector('#df-99933817-6409-4bc2-a013-b958d2078fe7 button.colab-df-convert');\n",
              "        buttonEl.style.display =\n",
              "          google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "        async function convertToInteractive(key) {\n",
              "          const element = document.querySelector('#df-99933817-6409-4bc2-a013-b958d2078fe7');\n",
              "          const dataTable =\n",
              "            await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                     [key], {});\n",
              "          if (!dataTable) return;\n",
              "\n",
              "          const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "            '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "            + ' to learn more about interactive tables.';\n",
              "          element.innerHTML = '';\n",
              "          dataTable['output_type'] = 'display_data';\n",
              "          await google.colab.output.renderOutput(dataTable, element);\n",
              "          const docLink = document.createElement('div');\n",
              "          docLink.innerHTML = docLinkHtml;\n",
              "          element.appendChild(docLink);\n",
              "        }\n",
              "      </script>\n",
              "    </div>\n",
              "  </div>\n",
              "  "
            ],
            "text/plain": [
              "             Всего ридов  Выровнилось уникально   Выровнилось неуникально  \\\n",
              "ENCFF019SLS      4697829                  100272                   119364   \n",
              "ENCFF641KVN      6467900                  140146                   145881   \n",
              "ENCFF411GHV      2349952                   80393                   162743   \n",
              "\n",
              "             Не выровнилось  \n",
              "ENCFF019SLS         4478193  \n",
              "ENCFF641KVN         6181873  \n",
              "ENCFF411GHV         2106816  "
            ]
          },
          "metadata": {},
          "execution_count": 19
        }
      ],
      "source": [
        "import pandas as pd\n",
        "data = {'Всего ридов':[4697829, 6467900, 2349952],\n",
        "        'Выровнилось уникально ':[100272, 140146, 80393],\n",
        "        'Выровнилось неуникально':[119364, 145881, 162743],\n",
        "        'Не выровнилось':[4478193, 6181873, 2106816]}\n",
        " \n",
        "frame = pd.DataFrame(data, index =['ENCFF019SLS',\n",
        "                                'ENCFF641KVN',\n",
        "                                'ENCFF411GHV'])\n",
        "frame"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "pTNt4qCXEuCf"
      },
      "source": [
        "Процент выравниваний получился низким, из за того что выравнивание ридов производилось на одну так какхромосому"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "TmfJKT0rmLyp"
      },
      "source": [
        "Преобразование .sam в .bam (не требуется если запускать macs2 на .sam файлах)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "2PteJ-KVqFhs"
      },
      "outputs": [],
      "source": [
        "# !samtools view -h -S -b -o bowtie2_res/res_ASU.bam bowtie2_res/res_ASU.sam\n",
        "# !samtools view -h -S -b -o bowtie2_res/res_ARK.bam bowtie2_res/res_ARK.sam"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "eLtG7xtRup19"
      },
      "source": [
        "На этом моменте можно скачать себе на компьютер все полученные .bam или .sam файлы на случай сбоя."
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "XZtQrPZMmgnC"
      },
      "source": [
        "## Peak calling\n",
        "\n",
        "Помимо macs2 может быть использован пакет Homer для поиска пиков, а также macs3."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 20,
      "metadata": {
        "id": "uX1ps4VDp_Ae"
      },
      "outputs": [],
      "source": [
        "!mkdir macs2"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 21,
      "metadata": {
        "id": "EJQOCcv2nJW_",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "f6865436-f7a1-40dc-9c1f-e6597057b13d"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "INFO  @ Thu, 10 Mar 2022 20:44:23: \n",
            "# Command line: callpeak --broad -t bowtie2_res/res_SLS.sam -c bowtie2_res/res_GHV.sam -f SAM --outdir macs2 -n ENCFF019SLS\n",
            "# ARGUMENTS LIST:\n",
            "# name = ENCFF019SLS\n",
            "# format = SAM\n",
            "# ChIP-seq file = ['bowtie2_res/res_SLS.sam']\n",
            "# control file = ['bowtie2_res/res_GHV.sam']\n",
            "# effective genome size = 2.70e+09\n",
            "# band width = 300\n",
            "# model fold = [5, 50]\n",
            "# qvalue cutoff for narrow/strong regions = 5.00e-02\n",
            "# qvalue cutoff for broad/weak regions = 1.00e-01\n",
            "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
            "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
            "# Larger dataset will be scaled towards smaller dataset.\n",
            "# Range for calculating regional lambda is: 1000 bps and 10000 bps\n",
            "# Broad region calling is on\n",
            "# Paired-End mode is off\n",
            " \n",
            "INFO  @ Thu, 10 Mar 2022 20:44:23: #1 read tag files... \n",
            "INFO  @ Thu, 10 Mar 2022 20:44:23: #1 read treatment tags... \n",
            "INFO  @ Thu, 10 Mar 2022 20:44:27: #1.2 read input tags... \n",
            "INFO  @ Thu, 10 Mar 2022 20:44:29: #1 tag size is determined as 76 bps \n",
            "INFO  @ Thu, 10 Mar 2022 20:44:29: #1 tag size = 76.0 \n",
            "INFO  @ Thu, 10 Mar 2022 20:44:29: #1  total tags in treatment: 219636 \n",
            "INFO  @ Thu, 10 Mar 2022 20:44:29: #1 user defined the maximum tags... \n",
            "INFO  @ Thu, 10 Mar 2022 20:44:29: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) \n",
            "INFO  @ Thu, 10 Mar 2022 20:44:29: #1  tags after filtering in treatment: 169829 \n",
            "INFO  @ Thu, 10 Mar 2022 20:44:29: #1  Redundant rate of treatment: 0.23 \n",
            "INFO  @ Thu, 10 Mar 2022 20:44:29: #1  total tags in control: 243136 \n",
            "INFO  @ Thu, 10 Mar 2022 20:44:29: #1 user defined the maximum tags... \n",
            "INFO  @ Thu, 10 Mar 2022 20:44:29: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) \n",
            "INFO  @ Thu, 10 Mar 2022 20:44:29: #1  tags after filtering in control: 197480 \n",
            "INFO  @ Thu, 10 Mar 2022 20:44:29: #1  Redundant rate of control: 0.19 \n",
            "INFO  @ Thu, 10 Mar 2022 20:44:29: #1 finished! \n",
            "INFO  @ Thu, 10 Mar 2022 20:44:29: #2 Build Peak Model... \n",
            "INFO  @ Thu, 10 Mar 2022 20:44:29: #2 looking for paired plus/minus strand peaks... \n",
            "INFO  @ Thu, 10 Mar 2022 20:44:29: #2 number of paired peaks: 1461 \n",
            "INFO  @ Thu, 10 Mar 2022 20:44:29: start model_add_line... \n",
            "INFO  @ Thu, 10 Mar 2022 20:44:29: start X-correlation... \n",
            "INFO  @ Thu, 10 Mar 2022 20:44:29: end of X-cor \n",
            "INFO  @ Thu, 10 Mar 2022 20:44:29: #2 finished! \n",
            "INFO  @ Thu, 10 Mar 2022 20:44:29: #2 predicted fragment length is 73 bps \n",
            "INFO  @ Thu, 10 Mar 2022 20:44:29: #2 alternative fragment length(s) may be 73,184,209,223,265,292 bps \n",
            "INFO  @ Thu, 10 Mar 2022 20:44:29: #2.2 Generate R script for model : macs2/ENCFF019SLS_model.r \n",
            "WARNING @ Thu, 10 Mar 2022 20:44:29: #2 Since the d (73) calculated from paired-peaks are smaller than 2*tag length, it may be influenced by unknown sequencing problem! \n",
            "WARNING @ Thu, 10 Mar 2022 20:44:29: #2 You may need to consider one of the other alternative d(s): 73,184,209,223,265,292 \n",
            "WARNING @ Thu, 10 Mar 2022 20:44:29: #2 You can restart the process with --nomodel --extsize XXX with your choice or an arbitrary number. Nontheless, MACS will continute computing. \n",
            "INFO  @ Thu, 10 Mar 2022 20:44:29: #3 Call peaks... \n",
            "INFO  @ Thu, 10 Mar 2022 20:44:29: #3 Call broad peaks with given level1 -log10qvalue cutoff and level2: 1.301030, 1.000000... \n",
            "INFO  @ Thu, 10 Mar 2022 20:44:29: #3 Pre-compute pvalue-qvalue table... \n",
            "INFO  @ Thu, 10 Mar 2022 20:44:30: #3 Call peaks for each chromosome... \n",
            "INFO  @ Thu, 10 Mar 2022 20:44:31: #4 Write output xls file... macs2/ENCFF019SLS_peaks.xls \n",
            "INFO  @ Thu, 10 Mar 2022 20:44:31: #4 Write broad peak in broadPeak format file... macs2/ENCFF019SLS_peaks.broadPeak \n",
            "INFO  @ Thu, 10 Mar 2022 20:44:31: #4 Write broad peak in bed12/gappedPeak format file... macs2/ENCFF019SLS_peaks.gappedPeak \n",
            "INFO  @ Thu, 10 Mar 2022 20:44:31: Done! \n"
          ]
        }
      ],
      "source": [
        "!macs2 callpeak --broad -t bowtie2_res/res_SLS.sam \\\n",
        "    -c bowtie2_res/res_GHV.sam \\\n",
        " \t  -f SAM \\\n",
        "\t  --outdir macs2 \\\n",
        "\t  -n ENCFF019SLS"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 22,
      "metadata": {
        "id": "xJPWujReF_zH",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "83ff87e2-bc46-4c86-c3e9-21eebcbf1d53"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "INFO  @ Thu, 10 Mar 2022 20:44:34: \n",
            "# Command line: callpeak --broad -t bowtie2_res/res_KVN.sam -c bowtie2_res/res_GHV.sam -f SAM --outdir macs2 -n ENCFF641KVN\n",
            "# ARGUMENTS LIST:\n",
            "# name = ENCFF641KVN\n",
            "# format = SAM\n",
            "# ChIP-seq file = ['bowtie2_res/res_KVN.sam']\n",
            "# control file = ['bowtie2_res/res_GHV.sam']\n",
            "# effective genome size = 2.70e+09\n",
            "# band width = 300\n",
            "# model fold = [5, 50]\n",
            "# qvalue cutoff for narrow/strong regions = 5.00e-02\n",
            "# qvalue cutoff for broad/weak regions = 1.00e-01\n",
            "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
            "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
            "# Larger dataset will be scaled towards smaller dataset.\n",
            "# Range for calculating regional lambda is: 1000 bps and 10000 bps\n",
            "# Broad region calling is on\n",
            "# Paired-End mode is off\n",
            " \n",
            "INFO  @ Thu, 10 Mar 2022 20:44:34: #1 read tag files... \n",
            "INFO  @ Thu, 10 Mar 2022 20:44:34: #1 read treatment tags... \n",
            "INFO  @ Thu, 10 Mar 2022 20:44:40: #1.2 read input tags... \n",
            "INFO  @ Thu, 10 Mar 2022 20:44:42: #1 tag size is determined as 76 bps \n",
            "INFO  @ Thu, 10 Mar 2022 20:44:42: #1 tag size = 76.0 \n",
            "INFO  @ Thu, 10 Mar 2022 20:44:42: #1  total tags in treatment: 286027 \n",
            "INFO  @ Thu, 10 Mar 2022 20:44:42: #1 user defined the maximum tags... \n",
            "INFO  @ Thu, 10 Mar 2022 20:44:42: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) \n",
            "INFO  @ Thu, 10 Mar 2022 20:44:42: #1  tags after filtering in treatment: 215384 \n",
            "INFO  @ Thu, 10 Mar 2022 20:44:42: #1  Redundant rate of treatment: 0.25 \n",
            "INFO  @ Thu, 10 Mar 2022 20:44:42: #1  total tags in control: 243136 \n",
            "INFO  @ Thu, 10 Mar 2022 20:44:42: #1 user defined the maximum tags... \n",
            "INFO  @ Thu, 10 Mar 2022 20:44:42: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) \n",
            "INFO  @ Thu, 10 Mar 2022 20:44:42: #1  tags after filtering in control: 197480 \n",
            "INFO  @ Thu, 10 Mar 2022 20:44:42: #1  Redundant rate of control: 0.19 \n",
            "INFO  @ Thu, 10 Mar 2022 20:44:42: #1 finished! \n",
            "INFO  @ Thu, 10 Mar 2022 20:44:42: #2 Build Peak Model... \n",
            "INFO  @ Thu, 10 Mar 2022 20:44:42: #2 looking for paired plus/minus strand peaks... \n",
            "INFO  @ Thu, 10 Mar 2022 20:44:42: #2 number of paired peaks: 1468 \n",
            "INFO  @ Thu, 10 Mar 2022 20:44:42: start model_add_line... \n",
            "INFO  @ Thu, 10 Mar 2022 20:44:42: start X-correlation... \n",
            "INFO  @ Thu, 10 Mar 2022 20:44:42: end of X-cor \n",
            "INFO  @ Thu, 10 Mar 2022 20:44:42: #2 finished! \n",
            "INFO  @ Thu, 10 Mar 2022 20:44:42: #2 predicted fragment length is 73 bps \n",
            "INFO  @ Thu, 10 Mar 2022 20:44:42: #2 alternative fragment length(s) may be 73,270,558 bps \n",
            "INFO  @ Thu, 10 Mar 2022 20:44:42: #2.2 Generate R script for model : macs2/ENCFF641KVN_model.r \n",
            "WARNING @ Thu, 10 Mar 2022 20:44:42: #2 Since the d (73) calculated from paired-peaks are smaller than 2*tag length, it may be influenced by unknown sequencing problem! \n",
            "WARNING @ Thu, 10 Mar 2022 20:44:42: #2 You may need to consider one of the other alternative d(s): 73,270,558 \n",
            "WARNING @ Thu, 10 Mar 2022 20:44:42: #2 You can restart the process with --nomodel --extsize XXX with your choice or an arbitrary number. Nontheless, MACS will continute computing. \n",
            "INFO  @ Thu, 10 Mar 2022 20:44:42: #3 Call peaks... \n",
            "INFO  @ Thu, 10 Mar 2022 20:44:42: #3 Call broad peaks with given level1 -log10qvalue cutoff and level2: 1.301030, 1.000000... \n",
            "INFO  @ Thu, 10 Mar 2022 20:44:42: #3 Pre-compute pvalue-qvalue table... \n",
            "INFO  @ Thu, 10 Mar 2022 20:44:43: #3 Call peaks for each chromosome... \n",
            "INFO  @ Thu, 10 Mar 2022 20:44:44: #4 Write output xls file... macs2/ENCFF641KVN_peaks.xls \n",
            "INFO  @ Thu, 10 Mar 2022 20:44:44: #4 Write broad peak in broadPeak format file... macs2/ENCFF641KVN_peaks.broadPeak \n",
            "INFO  @ Thu, 10 Mar 2022 20:44:44: #4 Write broad peak in bed12/gappedPeak format file... macs2/ENCFF641KVN_peaks.gappedPeak \n",
            "INFO  @ Thu, 10 Mar 2022 20:44:44: Done! \n"
          ]
        }
      ],
      "source": [
        "!macs2 callpeak --broad -t bowtie2_res/res_KVN.sam \\\n",
        "    -c bowtie2_res/res_GHV.sam \\\n",
        " \t  -f SAM \\\n",
        "\t  --outdir macs2 \\\n",
        "\t  -n ENCFF641KVN"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 23,
      "metadata": {
        "id": "HrNwxs7ArPz7",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "d15d1425-2565-4f4e-9154-ad89e09a63ed"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2022-03-10 20:44:48--  https://www.encodeproject.org/files/ENCFF546HBF/@@download/ENCFF546HBF.bed.gz\n",
            "Resolving www.encodeproject.org (www.encodeproject.org)... 34.211.244.144\n",
            "Connecting to www.encodeproject.org (www.encodeproject.org)|34.211.244.144|:443... connected.\n",
            "HTTP request sent, awaiting response... 307 Temporary Redirect\n",
            "Location: https://encode-public.s3.amazonaws.com/2021/05/05/9b56f0b0-0a2b-4169-a839-67332bc4fa35/ENCFF546HBF.bed.gz?response-content-disposition=attachment%3B%20filename%3DENCFF546HBF.bed.gz&AWSAccessKeyId=ASIATGZNGCNXZWYIVGBT&Signature=bepas7KSYdFIuwvP5mAOPjov0VU%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEIz%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FwEaCXVzLXdlc3QtMiJIMEYCIQDaY8r2KIMQ8ARjUrel4Z6jykxbuFMGVcD0w7VvEM1OiQIhAPKUaxlYciwOyweAaoEYFpKNivlgI1vuXAg8hxwiTzZmKoMECPX%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FwEQABoMMjIwNzQ4NzE0ODYzIgzBB7D2VncA0vaa8%2FQq1wPCjkTyUMOR5eVVmBJuyCcR0UYJcNEXpNK2e%2BJV9t9bUw6BryIvV7Yfn28SsUnpHlFQbvU7BSuPEkRQHuk3%2B%2BppyOMX%2BESx5EKkqOnRwBdSrQscAGJoxgkYSXgnkajWY6V4YZsHGY1%2FiufF259ZXCtApRqRuCLdxr7zC8069AAUKMT6fv5uqPb1C4tXsUrSBEwLvvyJ5Y1ttunrGwxih3fvx3b%2F1ZPqM4YSfL%2B25SqA0Wlz69PT7jwPOoCV2A94uSmROxTTf17kkqyMPEi3tat5nR7ZSdGsnueYAzFumnBWa7T2j51VqfQpjzUJDpY6ziJd2mMnPlILmCDXBA3C91aC%2FBJARmtjEksN7XXe6BdJVJlzfP7B8DRQdU6ck%2FISCgqLDWHFhZAA482Vz6hjcWaHIunrDu3qzl3eLY03dZZU9yx2V0kdE0ukl3W8RWjtOl%2B7Wa8WVeoIIrkEQDH%2FLqFNBXN8s3Wa7WDy%2F6D8w1jBdrlDdUB4D2kPtnvCZbR6J9TjdJ0DS0bc%2FrkxS3ktEd8p2SgSnD5ZoP4vE3OYQtCTkjqXjcVdf8BmUmpUTKhvrdGzLv9DU5BoiWn8w5DMz3KN%2Ber3BsqBWte8gKO54yg3w%2Bf3zRxlzekwobSpkQY6pAFPJozSjJNJo0xURHIS5Xe8bd13mNi8NaFYpe594jqzUQBGltmdkQeMBMy4TpcF%2BbXXW232jmHZHJAXhDggAaECiVOw7VzfuyErihSE1Q6R1g%2FGfcnLjYkC%2Be11vQWycDq81aku46koOKWZEtr3iomfG%2B8k9Zby3zDiuUMLJU5MN%2Fucr4OXqACbi7d6QtLORHMRQBSyZPNwVOBI1NFbC0xEocNFUA%3D%3D&Expires=1647074688 [following]\n",
            "--2022-03-10 20:44:48--  https://encode-public.s3.amazonaws.com/2021/05/05/9b56f0b0-0a2b-4169-a839-67332bc4fa35/ENCFF546HBF.bed.gz?response-content-disposition=attachment%3B%20filename%3DENCFF546HBF.bed.gz&AWSAccessKeyId=ASIATGZNGCNXZWYIVGBT&Signature=bepas7KSYdFIuwvP5mAOPjov0VU%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEIz%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FwEaCXVzLXdlc3QtMiJIMEYCIQDaY8r2KIMQ8ARjUrel4Z6jykxbuFMGVcD0w7VvEM1OiQIhAPKUaxlYciwOyweAaoEYFpKNivlgI1vuXAg8hxwiTzZmKoMECPX%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FwEQABoMMjIwNzQ4NzE0ODYzIgzBB7D2VncA0vaa8%2FQq1wPCjkTyUMOR5eVVmBJuyCcR0UYJcNEXpNK2e%2BJV9t9bUw6BryIvV7Yfn28SsUnpHlFQbvU7BSuPEkRQHuk3%2B%2BppyOMX%2BESx5EKkqOnRwBdSrQscAGJoxgkYSXgnkajWY6V4YZsHGY1%2FiufF259ZXCtApRqRuCLdxr7zC8069AAUKMT6fv5uqPb1C4tXsUrSBEwLvvyJ5Y1ttunrGwxih3fvx3b%2F1ZPqM4YSfL%2B25SqA0Wlz69PT7jwPOoCV2A94uSmROxTTf17kkqyMPEi3tat5nR7ZSdGsnueYAzFumnBWa7T2j51VqfQpjzUJDpY6ziJd2mMnPlILmCDXBA3C91aC%2FBJARmtjEksN7XXe6BdJVJlzfP7B8DRQdU6ck%2FISCgqLDWHFhZAA482Vz6hjcWaHIunrDu3qzl3eLY03dZZU9yx2V0kdE0ukl3W8RWjtOl%2B7Wa8WVeoIIrkEQDH%2FLqFNBXN8s3Wa7WDy%2F6D8w1jBdrlDdUB4D2kPtnvCZbR6J9TjdJ0DS0bc%2FrkxS3ktEd8p2SgSnD5ZoP4vE3OYQtCTkjqXjcVdf8BmUmpUTKhvrdGzLv9DU5BoiWn8w5DMz3KN%2Ber3BsqBWte8gKO54yg3w%2Bf3zRxlzekwobSpkQY6pAFPJozSjJNJo0xURHIS5Xe8bd13mNi8NaFYpe594jqzUQBGltmdkQeMBMy4TpcF%2BbXXW232jmHZHJAXhDggAaECiVOw7VzfuyErihSE1Q6R1g%2FGfcnLjYkC%2Be11vQWycDq81aku46koOKWZEtr3iomfG%2B8k9Zby3zDiuUMLJU5MN%2Fucr4OXqACbi7d6QtLORHMRQBSyZPNwVOBI1NFbC0xEocNFUA%3D%3D&Expires=1647074688\n",
            "Resolving encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)... 52.218.249.243\n",
            "Connecting to encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)|52.218.249.243|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 2849215 (2.7M) [binary/octet-stream]\n",
            "Saving to: ‘ENCFF546HBF.bed.gz’\n",
            "\n",
            "ENCFF546HBF.bed.gz  100%[===================>]   2.72M  4.76MB/s    in 0.6s    \n",
            "\n",
            "2022-03-10 20:44:49 (4.76 MB/s) - ‘ENCFF546HBF.bed.gz’ saved [2849215/2849215]\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!wget https://www.encodeproject.org/files/ENCFF546HBF/@@download/ENCFF546HBF.bed.gz\n",
        "!gzip -d ENCFF546HBF.bed.gz"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "MYEGEbQjvEwY"
      },
      "source": [
        "## Сравнение результатов\n",
        "\n",
        "Сравниваем те пики, которые мы получили, с пиками, которые приведены в ENCODE (важно, чтобы версии генома hg38 или hg19) для .bed файла из ENCODE и той хромосомы, которую скачивали выше, сопадали.\n",
        "\n",
        "Проанализируйте полученные результаты и приведите свои рассуждения в README.md. Как можно объяснить различия в количестве пересечений?"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 24,
      "metadata": {
        "id": "Hi6qQ8jkJ_cf",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "1414f073-59f7-4cf6-cf18-e8d90ee25118"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "Generating a 2-way \"venn\" diagram. Please wait...\n",
            "\n",
            "\n",
            "Done! Please check your results @ /content/Intervene_results. \n",
            "Thank you for using Intervene!\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!intervene venn -i macs2/ENCFF019SLS_peaks.broadPeak ENCFF546HBF.bed  --filenames "
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 25,
      "metadata": {
        "id": "lCCxC-XUK3Sa",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "0fd1ba3c-fec5-4d7b-8815-661869c4d660"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "Generating a 2-way \"venn\" diagram. Please wait...\n",
            "\n",
            "\n",
            "Done! Please check your results @ /content/Intervene_results. \n",
            "Thank you for using Intervene!\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!intervene venn -i ENCFF546HBF.bed macs2/ENCFF019SLS_peaks.broadPeak  --filenames"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 26,
      "metadata": {
        "id": "7fV_30xCLTAS",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "16f20567-c83a-4ca7-bd83-44db0e186044"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "Generating a 2-way \"venn\" diagram. Please wait...\n",
            "\n",
            "\n",
            "Done! Please check your results @ /content/Intervene_results. \n",
            "Thank you for using Intervene!\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!intervene venn -i macs2/ENCFF641KVN_peaks.broadPeak ENCFF546HBF.bed  --filenames"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 27,
      "metadata": {
        "id": "Dgg-ONfdRFm7",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "0d3e044e-53ca-4c67-f6da-70a09e1f6ae5"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "Generating a 2-way \"venn\" diagram. Please wait...\n",
            "\n",
            "\n",
            "Done! Please check your results @ /content/Intervene_results. \n",
            "Thank you for using Intervene!\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!intervene venn -i ENCFF546HBF.bed macs2/ENCFF641KVN_peaks.broadPeak  --filenames"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "isYx230PAANq"
      },
      "outputs": [],
      "source": [
        ""
      ]
    }
  ],
  "metadata": {
    "colab": {
      "collapsed_sections": [],
      "name": "Copy_of_ChIP_seq_on_histones-Copy1.ipynb",
      "provenance": []
    },
    "kernelspec": {
      "display_name": "Python 3",
      "name": "python3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 0
}